In this training, you will:

### **1. :warning: Outside of class hours**
1. **Access your Google Cloud account using your academic email address**

    You have been provided with Google Cloud Education coupons to comfortably
    run a Google Cloud Engine (GCE) Virtual Machine (VM).

2. **Deploy your own Galaxy server by launching a Virtual Machine (VM) from a Virtual Machine Image (VMI)**
3. **Register to your Galaxy server as an admin.** and post an artifact showing that you actually did it.

![](images/checkpoint.png){width="70px"}

4. **Review a few command line useful to manage you Galaxy server** (using an ssh session in your VM).
    
### **2. During the class session of Monday afternoon (December 1st)**
1. **Restart you VM with your Galaxy server**
2. **Check that you Galaxy server is up and running properly**
3. **Transfer the needed data from the data library to Galaxy histories.**
      
      This includes genome reference indexes, reference data (RNA classes, GTF files, etc.),
      and sequencing datasets.

4. **Compare the procedures for performing a simple file formatting task**, using either command lines (Linux environment) or the Galaxy interface.

By the end of this training, you should be familiar with the Galaxy analysis
environment and ready to perform analyses of small RNA-seq and RNA-seq datasets,
as described [here](https://slecrom.github.io/AG2025/).