
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://artbio.github.io/startbio/scRNAseq_basics/import/">
      
      
        <link rel="prev" href="../intro_seurat/">
      
      
        <link rel="next" href="../preprocessing/">
      
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.5.3, mkdocs-material-9.5.17">
    
    
      
        <title>Import data - STARTbio</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.bcfcd587.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.06af60db.min.css">
      
      
  
  
    
    
  
  
  <style>:root{--md-admonition-icon--danger:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 12"><path d="M4.855.708c.5-.896 1.79-.896 2.29 0l4.675 8.351a1.312 1.312 0 0 1-1.146 1.954H1.33A1.313 1.313 0 0 1 .183 9.058ZM7 7V3H5v4Zm-1 3a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z"/></svg>');}</style>



    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Noto+Sans+Display:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Noto Sans Display";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../spoiler_files/spoiler.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="black" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#import-data" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="STARTbio" class="md-header__button md-logo" aria-label="STARTbio" data-md-component="logo">
      
  <img src="../../assets/startbio_icon.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            STARTbio
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Import data
            
          </span>
        </div>
      </div>
    </div>
    
      
    
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/ARTbio/startbio" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    ARTbio/startbio
  </div>
</a>
      </div>
    
  </nav>
  
    
      
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../.." class="md-tabs__link">
        
  
    
  
  Home

      </a>
    </li>
  

      
        
  
  
    
  
  
    
    
      
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="../../bulk_RNAseq-IOC/00_IOC_RNAseq_program/" class="md-tabs__link">
          
  
  Interactive Online Companionship (IOC)

        </a>
      </li>
    
  

    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../ten2/Regex_01/" class="md-tabs__link">
          
  
  1Omin2

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../AnalyseGenomes_2025/outline/" class="md-tabs__link">
          
  
  Galaxy Trainings

        </a>
      </li>
    
  

    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../bash_snippets/" class="md-tabs__link">
          
  
  Code Snippets

        </a>
      </li>
    
  

      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../../links/" class="md-tabs__link">
        
  
    
  
  Links

      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
    
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="STARTbio" class="md-nav__button md-logo" aria-label="STARTbio" data-md-component="logo">
      
  <img src="../../assets/startbio_icon.png" alt="logo">

    </a>
    STARTbio
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/ARTbio/startbio" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    ARTbio/startbio
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Home
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
      
        
        
      
      
    
    
      
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" checked>
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="">
            
  
  <span class="md-ellipsis">
    Interactive Online Companionship (IOC)
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            Interactive Online Companionship (IOC)
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../../bulk_RNAseq-IOC/00_IOC_RNAseq_program/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    IOC on bulk RNAseq analysis
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    
  
  
    <a href="../../R-IOC/00_IOC_R_program/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    IOC on R
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

  

      
    </li>
  

              
            
              
                
  
  
    
  
  
  
    
    
    
      
      
    
    
      
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_3" checked>
        
          
          <label class="md-nav__link" for="__nav_2_3" id="__nav_2_3_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    IOC on Single-cell RNAseq analysis using Seurat in R
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_3_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_2_3">
            <span class="md-nav__icon md-icon"></span>
            IOC on Single-cell RNAseq analysis using Seurat in R
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../program_sc/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Program / Schedule
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

              
            
              
                
  
  
    
  
  
  
    
    
    
      
      
    
    
      
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_3_2" checked>
        
          
          <label class="md-nav__link" for="__nav_2_3_2" id="__nav_2_3_2_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Reference Manual
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_2_3_2_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_2_3_2">
            <span class="md-nav__icon md-icon"></span>
            Reference Manual
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../introduction/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Introduction
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../prerequisites/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Prerequisites
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../intro_seurat/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Initialization of R analysis
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    Import data
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    Import data
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#expression-matrix" class="md-nav__link">
    <span class="md-ellipsis">
      Expression matrix
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#dataset-test" class="md-nav__link">
    <span class="md-ellipsis">
      Dataset test
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#biomart-annotation" class="md-nav__link">
    <span class="md-ellipsis">
      biomaRt Annotation
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../preprocessing/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Preprocessing
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../redim/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Reduction of dimensionality
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../clustering/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Clustering
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../intro_markers/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Cell population identification
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../comp/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Comparison between 2 populations
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../visualization/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Differents visualizations of Seurat
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../references/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    References
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../../deepseq/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Introduction to Deep-seq Analysis
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../../ten2/Regex_01/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    1Omin2
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    
  
  
    <a href="../../AnalyseGenomes_2025/outline/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Galaxy Trainings
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

  

      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../../bash_snippets/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Code Snippets
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../links/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Links
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#expression-matrix" class="md-nav__link">
    <span class="md-ellipsis">
      Expression matrix
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#dataset-test" class="md-nav__link">
    <span class="md-ellipsis">
      Dataset test
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#biomart-annotation" class="md-nav__link">
    <span class="md-ellipsis">
      biomaRt Annotation
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
    <a href="https://github.com/ARTbio/startbio/edit/master/docs/scRNAseq_basics/import.md" title="Edit this page" class="md-content__button md-icon">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M10 20H6V4h7v5h5v3.1l2-2V8l-6-6H6c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h4v-2m10.2-7c.1 0 .3.1.4.2l1.3 1.3c.2.2.2.6 0 .8l-1 1-2.1-2.1 1-1c.1-.1.2-.2.4-.2m0 3.9L14.1 23H12v-2.1l6.1-6.1 2.1 2.1Z"/></svg>
    </a>
  
  
    
      
    
    <a href="https://github.com/ARTbio/startbio/raw/master/docs/scRNAseq_basics/import.md" title="View source of this page" class="md-content__button md-icon">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 18c.56 0 1 .44 1 1s-.44 1-1 1-1-.44-1-1 .44-1 1-1m0-3c-2.73 0-5.06 1.66-6 4 .94 2.34 3.27 4 6 4s5.06-1.66 6-4c-.94-2.34-3.27-4-6-4m0 6.5a2.5 2.5 0 0 1-2.5-2.5 2.5 2.5 0 0 1 2.5-2.5 2.5 2.5 0 0 1 2.5 2.5 2.5 2.5 0 0 1-2.5 2.5M9.27 20H6V4h7v5h5v4.07c.7.08 1.36.25 2 .49V8l-6-6H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h4.5a8.15 8.15 0 0 1-1.23-2Z"/></svg>
    </a>
  


<h1 id="import-data">Import Data<a class="headerlink" href="#import-data" title="Permanent link">&para;</a></h1>
<h2 id="expression-matrix">Expression matrix<a class="headerlink" href="#expression-matrix" title="Permanent link">&para;</a></h2>
<p>The first thing to import is the expression data. They are
generally in the form of 3 files:</p>
<ul>
<li>The name of the transcriptomes <code>barcodes.tsv</code> : file with one column with the
  transcriptome names (cells, barcodes)</li>
<li>Gene names in <code>genes.tsv</code> format : three columns file
  (geneID of the used gtf annotation, gene name, "Gene Expression")</li>
<li>the expression matrix <code>matrix.mtx</code> : three columns file
  (position of the transcriptome in the <code>barcodes.tsv</code> file, position of the of
  the gene in the file <code>genes.tsv</code>, associated expression level)</li>
</ul>
<p>The <code>Read10X</code> function of Seurat allows to create from these files the
expression matrix in <code>dgCMatrix</code> format. It is a sparse matrix used for
efficient processing of large matrices.</p>
<div class="admonition danger">
<p class="admonition-title">Caution</p>
<p>A very important parameter of <code>Read10X</code> is <code>gene.column</code> which allows us
to choose which column of the <code>genes.tsv</code> file to use. It is preferable
to select the column composed only of gene ID during the analysis (column 1)
and to use the gene names only during the annotation of the marker genes.
Indeed column 2 of <code>genes.tsv</code> is in fact composed of gene name and
<em>gene ID</em> (because there is not necessarily a gene name assigned to each
unique identifier). Moreover, several <em>gene ID</em> can correspond to a single
gene name making the annotation step very complex.
If we choose column 2, duplicates in gene names are handled with the
<code>make.unique</code> function which detects duplicates and then adds
<code>.1, .2, ..., .n</code> after each occurrence (knowing that the first occurrence
is not modified). With these modified gene names, it will be very difficult
to use the different databases that will not be able to recognize textually
these new gene names.</p>
<p>We could think that it would be easy to find the genes whose names have
been modified afterwards by searching all the genes that would have a dot
in their names, unfortunately some gene names already contain dots making
the search for patterns even more complex. To get rid of all these
problems, I <strong>strongly recommend</strong> to use the first column and to use only
gene names afterwards.</p>
</div>
<h2 id="dataset-test">Dataset test<a class="headerlink" href="#dataset-test" title="Permanent link">&para;</a></h2>
<p>For this usecase, the dataset used for the analysis is composed of peripheral 
blood mononuclear cells (PBMC). It is available on the 10X and Seurat website 
which can be found on the <a href="https://satijalab.org/seurat/articles/pbmc3k_tutorial.html">Seurat tutorial</a>. The first thing to do is to dowload the dataset directly from 10X website. 
We use the R function <code>system</code> to execute command lines that we normally run in a 
terminal. Then we use the function <code>Read10X</code> to read the bundle format that are the
direct output of CellRanger (with specific filenames). If you have another type of
bundle format, you might be interessed by the function <code>ReadMtx</code>.</p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-0-1"><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a><span class="c1">## Download expression matrix from 10X website in my subfolder &quot;test_data&quot;</span>
</span><span id="__span-0-2"><a id="__codelineno-0-2" name="__codelineno-0-2" href="#__codelineno-0-2"></a><span class="nf">system</span><span class="p">(</span><span class="s">&quot;wget -P ./test-data/ https://cf.10xgenomics.com/samples/cell/pbmc3k/pbmc3k_filtered_gene_bc_matrices.tar.gz&quot;</span><span class="p">)</span>
</span><span id="__span-0-3"><a id="__codelineno-0-3" name="__codelineno-0-3" href="#__codelineno-0-3"></a><span class="c1">## Extract archive in my subfolder &quot;test_data&quot;</span>
</span><span id="__span-0-4"><a id="__codelineno-0-4" name="__codelineno-0-4" href="#__codelineno-0-4"></a><span class="nf">system</span><span class="p">(</span><span class="s">&quot;tar -zxvf ./test-data/pbmc3k_filtered_gene_bc_matrices.tar.gz -C ./test-data/&quot;</span><span class="p">)</span>
</span><span id="__span-0-5"><a id="__codelineno-0-5" name="__codelineno-0-5" href="#__codelineno-0-5"></a>
</span><span id="__span-0-6"><a id="__codelineno-0-6" name="__codelineno-0-6" href="#__codelineno-0-6"></a><span class="c1">## Import expression matrix in R global environment</span>
</span><span id="__span-0-7"><a id="__codelineno-0-7" name="__codelineno-0-7" href="#__codelineno-0-7"></a><span class="n">tenX_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">Read10X</span><span class="p">(</span><span class="n">data.dir</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;./test-data/filtered_gene_bc_matrices/hg19&quot;</span><span class="p">,</span><span class="w"> </span><span class="c1">#Path to the directory containing the three 10X files (matrix.mtx, genes.tsv (or features.tsv depending on the CellRanger version), barcodes.tsv)</span>
</span><span id="__span-0-8"><a id="__codelineno-0-8" name="__codelineno-0-8" href="#__codelineno-0-8"></a><span class="w">                       </span><span class="n">gene.column</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="w"> </span><span class="c1">#Choice of the genes.tsv column that will determine the names of the genes in the expression matrix</span>
</span><span id="__span-0-9"><a id="__codelineno-0-9" name="__codelineno-0-9" href="#__codelineno-0-9"></a>
</span><span id="__span-0-10"><a id="__codelineno-0-10" name="__codelineno-0-10" href="#__codelineno-0-10"></a><span class="c1">## Matrix dimension : first value is the number of row (genes) and second value is the number of column (sample/barcodes)</span>
</span><span id="__span-0-11"><a id="__codelineno-0-11" name="__codelineno-0-11" href="#__codelineno-0-11"></a><span class="nf">dim</span><span class="p">(</span><span class="n">tenX_matrix</span><span class="p">)</span>
</span></code></pre></div>
<p>Description of the <code>dgCmatrix</code> object:</p>
<ul>
<li><code>i</code>: row position of each non-zero value of your
    expression matrix (knowing that the index of the row starts at
    zero)</li>
<li><code>p</code> : vector of size <code>number of barcode + 1</code> and is described as
    representing the number of non-zero values in each column (<code>j</code>)
    such that: <code>tenX_matrix@p[j+2] - tenX_matrix@p[j+1]</code></li>
<li><code>Dim</code> : vector containing the dimensions of your expression matrix
    expression matrix (number of genes, rows then number of barcodes,
    columns)</li>
<li><code>Dimnames</code> : list containing the names of the genes (rownames) and the
    barcode names (colnames)</li>
<li><code>x</code> : vector containing the values of non-null expressions</li>
<li><code>factors</code>.</li>
</ul>
<p>The most important thing to remember about this complex object is the slots
corresponding to the number of dimensions and their names (to know the
annotation of your genes and your barcodes) and <code>x</code> which allows to visualize
the different expression values. Useful to know if what you have in your hands
are integer values, so probably raw expression levels, or if you have decimal
values which would reflect normalized expression levels.</p>
<p>A <code>dgCMatrix</code> is displayed in the console just like a dataframe where the
points correspond to zero:</p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-1-1"><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a><span class="c1">## Visualize first three rows and columns</span>
</span><span id="__span-1-2"><a id="__codelineno-1-2" name="__codelineno-1-2" href="#__codelineno-1-2"></a><span class="n">tenX_matrix</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">3</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="m">3</span><span class="p">]</span>
</span></code></pre></div>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span>
<span class="normal">2</span>
<span class="normal">3</span>
<span class="normal">4</span>
<span class="normal">5</span></pre></div></td><td class="code"><div><pre><span></span><code>## 3 x 3 sparse Matrix of class &quot;dgCMatrix&quot;
##                 AAACATACAACCAC-1 AAACATTGAGCTAC-1 AAACATTGATCAGC-1
## ENSG00000243485                .                .                .
## ENSG00000237613                .                .                .
## ENSG00000186092                .                .                .
</code></pre></div></td></tr></table></div>

<div class="admonition question">
<p class="admonition-title">What if I don't have a bundle format ?</p>
<p>If the format of the expression matrix is not a three-file format but a
file containing a table (<em>n</em> genes x <em>n</em> barcodes), we can go directly to 
the next step. You just need to import it in your R global environment as 
a <code>matrix</code> or <code>data.frame</code> for example. Don't forget that the expression 
table must have barcodes as columns and genes as rows.</p>
</div>
<h2 id="biomart-annotation">biomaRt Annotation<a class="headerlink" href="#biomart-annotation" title="Permanent link">&para;</a></h2>
<p>We will then import the biomart database which will be used to annotate the
genes of our analysis. This will allow us to go from a gene ID set to a
gene name. The data of Seurat being in hg19 we recover the database for
this annotation.  </p>
<p>The biomaRt package allows us to retrieve the databases that are available
via <a href="https://www.ensembl.org/biomart/martview/aeb467155c9aeccaa44a70171cde4e15">Ensembl BioMart</a>
and to interact directly in the R environment.
The first function we will use is <code>useEnsembl</code> to connect to Ensembl. You can
see all the versions and genomes available through the function of the same
package <code>listEnsembl</code>. The result of <code>useEnsembl</code> is an S4 object of class
Mart which contains the database of the genome and the desired version.
A second step will allow us to obtain a dataframe with the annotation of the
genes. The <code>getBM</code> function will perform a query via the Mart object that we
have previously generated. It needs to be provided with a vector containing
different attributes (chromosome, gene name, homologs, orthologs, and many
others...). We can have the exhaustive list with the <code>listAttributes</code> function.</p>
<p>Here we will also filter to get the annotation only for our list of genes with
the parameters <em>filters</em> and <em>values</em>.  </p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-2-1"><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a><span class="c1">## Import of the biomart database for hg19</span>
</span><span id="__span-2-2"><a id="__codelineno-2-2" name="__codelineno-2-2" href="#__codelineno-2-2"></a><span class="n">ensembl_hg19</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">useEnsembl</span><span class="p">(</span><span class="n">biomart</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;genes&quot;</span><span class="p">,</span><span class="w">                  </span><span class="c1">#Import ensembl genes database</span>
</span><span id="__span-2-3"><a id="__codelineno-2-3" name="__codelineno-2-3" href="#__codelineno-2-3"></a><span class="w">                           </span><span class="n">dataset</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;hsapiens_gene_ensembl&quot;</span><span class="p">,</span><span class="w">  </span><span class="c1">#Genome</span>
</span><span id="__span-2-4"><a id="__codelineno-2-4" name="__codelineno-2-4" href="#__codelineno-2-4"></a><span class="w">                           </span><span class="n">GRCh</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">37</span><span class="p">)</span><span class="w">                          </span><span class="c1">#Genome version, only 37 or NULL for 38 are accepted for now</span>
</span><span id="__span-2-5"><a id="__codelineno-2-5" name="__codelineno-2-5" href="#__codelineno-2-5"></a>
</span><span id="__span-2-6"><a id="__codelineno-2-6" name="__codelineno-2-6" href="#__codelineno-2-6"></a><span class="c1">## If you don&#39;t know what to give to biomart parameter of useEnsembl just run :</span>
</span><span id="__span-2-7"><a id="__codelineno-2-7" name="__codelineno-2-7" href="#__codelineno-2-7"></a><span class="c1">#listEnsembl() #dataframe where the first column is the value to give to biomart parameter of useEnsembl</span>
</span><span id="__span-2-8"><a id="__codelineno-2-8" name="__codelineno-2-8" href="#__codelineno-2-8"></a>
</span><span id="__span-2-9"><a id="__codelineno-2-9" name="__codelineno-2-9" href="#__codelineno-2-9"></a><span class="c1">## If you don&#39;t know the available datasets :</span>
</span><span id="__span-2-10"><a id="__codelineno-2-10" name="__codelineno-2-10" href="#__codelineno-2-10"></a><span class="c1">#listDatasets(mart = useEnsembl(biomart = &quot;genes&quot;)) #dataframe where the first column (dataset) contains the value to give to dataset parameter of useEnsembl, add GRCh parameter in the useEnsembl function to precise your preferred genome version</span>
</span><span id="__span-2-11"><a id="__codelineno-2-11" name="__codelineno-2-11" href="#__codelineno-2-11"></a>
</span><span id="__span-2-12"><a id="__codelineno-2-12" name="__codelineno-2-12" href="#__codelineno-2-12"></a><span class="c1">## Recovering attributes according to our gene list (genes present in our expression matrix)</span>
</span><span id="__span-2-13"><a id="__codelineno-2-13" name="__codelineno-2-13" href="#__codelineno-2-13"></a><span class="n">annotated_hg19</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">getBM</span><span class="p">(</span><span class="n">attributes</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;ensembl_gene_id&quot;</span><span class="p">,</span>
</span><span id="__span-2-14"><a id="__codelineno-2-14" name="__codelineno-2-14" href="#__codelineno-2-14"></a><span class="w">                                       </span><span class="s">&quot;external_gene_name&quot;</span><span class="p">,</span>
</span><span id="__span-2-15"><a id="__codelineno-2-15" name="__codelineno-2-15" href="#__codelineno-2-15"></a><span class="w">                                       </span><span class="s">&quot;description&quot;</span><span class="p">,</span>
</span><span id="__span-2-16"><a id="__codelineno-2-16" name="__codelineno-2-16" href="#__codelineno-2-16"></a><span class="w">                                       </span><span class="s">&quot;gene_biotype&quot;</span><span class="p">,</span>
</span><span id="__span-2-17"><a id="__codelineno-2-17" name="__codelineno-2-17" href="#__codelineno-2-17"></a><span class="w">                                       </span><span class="s">&quot;chromosome_name&quot;</span><span class="p">),</span><span class="w">    </span><span class="c1">#Informations to retrieve</span>
</span><span id="__span-2-18"><a id="__codelineno-2-18" name="__codelineno-2-18" href="#__codelineno-2-18"></a><span class="w">                           </span><span class="n">filters</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ensembl_gene_id&quot;</span><span class="p">,</span><span class="w">       </span><span class="c1">#Which variable to choose for the filtering</span>
</span><span id="__span-2-19"><a id="__codelineno-2-19" name="__codelineno-2-19" href="#__codelineno-2-19"></a><span class="w">                           </span><span class="n">values</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">tenX_matrix</span><span class="p">),</span><span class="w">    </span><span class="c1">#Values that will filter the database</span>
</span><span id="__span-2-20"><a id="__codelineno-2-20" name="__codelineno-2-20" href="#__codelineno-2-20"></a><span class="w">                           </span><span class="n">mart</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ensembl_hg19</span><span class="p">)</span><span class="w">               </span><span class="c1">#Database</span>
</span><span id="__span-2-21"><a id="__codelineno-2-21" name="__codelineno-2-21" href="#__codelineno-2-21"></a>
</span><span id="__span-2-22"><a id="__codelineno-2-22" name="__codelineno-2-22" href="#__codelineno-2-22"></a><span class="c1">## If you need to know all available attributes and their name</span>
</span><span id="__span-2-23"><a id="__codelineno-2-23" name="__codelineno-2-23" href="#__codelineno-2-23"></a><span class="c1">#listAttributes(mart = ensembl_hg19) #dataframe where the first column is the attribute&#39; names needed for the attributes parameter of getBM</span>
</span><span id="__span-2-24"><a id="__codelineno-2-24" name="__codelineno-2-24" href="#__codelineno-2-24"></a>
</span><span id="__span-2-25"><a id="__codelineno-2-25" name="__codelineno-2-25" href="#__codelineno-2-25"></a><span class="c1">## Preview of the resulting dataframe</span>
</span><span id="__span-2-26"><a id="__codelineno-2-26" name="__codelineno-2-26" href="#__codelineno-2-26"></a><span class="nf">kable</span><span class="p">(</span><span class="nf">head</span><span class="p">(</span><span class="n">annotated_hg19</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;simple&quot;</span><span class="p">)</span>
</span></code></pre></div>
<table>
<thead>
<tr>
<th style="text-align: left;">ensembl_gene_id</th>
<th style="text-align: left;">external_gene_name</th>
<th style="text-align: left;">description</th>
<th style="text-align: left;">gene_biotype</th>
<th style="text-align: left;">chromosome_name</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;">ENSG00000000457</td>
<td style="text-align: left;">SCYL3</td>
<td style="text-align: left;">SCY1-like 3 (S. cerevisiae) [Source:HGNC Symbol;Acc:19285]</td>
<td style="text-align: left;">protein_coding</td>
<td style="text-align: left;">1</td>
</tr>
<tr>
<td style="text-align: left;">ENSG00000000460</td>
<td style="text-align: left;">C1orf112</td>
<td style="text-align: left;">chromosome 1 open reading frame 112 [Source:HGNC Symbol;Acc:25565]</td>
<td style="text-align: left;">protein_coding</td>
<td style="text-align: left;">1</td>
</tr>
<tr>
<td style="text-align: left;">ENSG00000000938</td>
<td style="text-align: left;">FGR</td>
<td style="text-align: left;">feline Gardner-Rasheed sarcoma viral oncogene homolog [Source:HGNC Symbol;Acc:3697]</td>
<td style="text-align: left;">protein_coding</td>
<td style="text-align: left;">1</td>
</tr>
<tr>
<td style="text-align: left;">ENSG00000000971</td>
<td style="text-align: left;">CFH</td>
<td style="text-align: left;">complement factor H [Source:HGNC Symbol;Acc:4883]</td>
<td style="text-align: left;">protein_coding</td>
<td style="text-align: left;">1</td>
</tr>
<tr>
<td style="text-align: left;">ENSG00000001460</td>
<td style="text-align: left;">STPG1</td>
<td style="text-align: left;">sperm-tail PG-rich repeat containing 1 [Source:HGNC Symbol;Acc:28070]</td>
<td style="text-align: left;">protein_coding</td>
<td style="text-align: left;">1</td>
</tr>
<tr>
<td style="text-align: left;">ENSG00000001461</td>
<td style="text-align: left;">NIPAL3</td>
<td style="text-align: left;">NIPA-like domain containing 3 [Source:HGNC Symbol;Acc:25233]</td>
<td style="text-align: left;">protein_coding</td>
<td style="text-align: left;">1</td>
</tr>
</tbody>
</table>
<p>We obtained a dataframe with the information about the genes present
in our expression matrix.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The <code>kable</code> function is only useful when you are coding in RMarkdown notebooks. 
It helps improve the table display. If you code in the R console or with a Rscript
you just need to execute <code>head(annotated_hg19)</code>.</p>
</div>
<h1 id="creation-of-the-seurat-object">Creation of the Seurat object<a class="headerlink" href="#creation-of-the-seurat-object" title="Permanent link">&para;</a></h1>
<p>Then thanks to the <code>CreateSeuratObject</code> function we can import the expression
matrix in a Seurat object which will be the basis of our analysis. All the
different steps will be associated to this object which is an S4 object
(tree of other smaller objects, dataframe, vectors...), specific to Seurat
called a <code>SeuratObject</code>.  </p>
<p>There are three important parameters:</p>
<ul>
<li><code>project</code>: the name of your project which will also be the primary identity
   of your cells. It is therefore very important to define a project name and
   not to leave the default value.</li>
<li><code>min.cells</code> : allows to filter out genes that are not detected in at least
  <code>min.cells</code>.</li>
<li><code>min.features</code> : allows to filter the cells which do not detect at least
  <code>min.features</code>. We will define <code>min.features = 1</code> to filter the barcodes
  (cells, nuclei) that do not contain any UMI. This also allows to reduce the weight
  of the Seurat object in our environment.</li>
</ul>
<div class="language-r highlight"><pre><span></span><code><span id="__span-3-1"><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a><span class="c1">## Creation of the Seurat Object</span>
</span><span id="__span-3-2"><a id="__codelineno-3-2" name="__codelineno-3-2" href="#__codelineno-3-2"></a><span class="n">pbmc_small</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">CreateSeuratObject</span><span class="p">(</span><span class="n">tenX_matrix</span><span class="p">,</span><span class="w">                    </span><span class="c1">#Expression matrix</span>
</span><span id="__span-3-3"><a id="__codelineno-3-3" name="__codelineno-3-3" href="#__codelineno-3-3"></a><span class="w">                                 </span><span class="n">project</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;PBMC analysis&quot;</span><span class="p">,</span><span class="w">      </span><span class="c1">#Name of the project : something meaningful for your dataset</span>
</span><span id="__span-3-4"><a id="__codelineno-3-4" name="__codelineno-3-4" href="#__codelineno-3-4"></a><span class="w">                                 </span><span class="n">assay</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;RNA&quot;</span><span class="p">,</span><span class="w">                  </span><span class="c1">#Name of the initial assay, (others can be created during the analysis), default is RNA</span>
</span><span id="__span-3-5"><a id="__codelineno-3-5" name="__codelineno-3-5" href="#__codelineno-3-5"></a><span class="w">                                 </span><span class="n">names.field</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w">                </span><span class="c1">#Associated with the names.delim, it allows to separate the name of the barcode when this one is composed of several information ex: BARCODE_CLUSTER_CELLTYPE and to choose after split on the names.delim which part we choose as the barcode name</span>
</span><span id="__span-3-6"><a id="__codelineno-3-6" name="__codelineno-3-6" href="#__codelineno-3-6"></a><span class="w">                                 </span><span class="n">names.delim</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;_&quot;</span><span class="p">,</span><span class="w">              </span><span class="c1">#Character that allows to dissociate the different information contained in the barcode names</span>
</span><span id="__span-3-7"><a id="__codelineno-3-7" name="__codelineno-3-7" href="#__codelineno-3-7"></a><span class="w">                                 </span><span class="n">meta.data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NULL</span><span class="p">,</span><span class="w">               </span><span class="c1">#We can add the metadata on the transcriptomes with a dataframe where the barcodes are in line and the different information in column</span>
</span><span id="__span-3-8"><a id="__codelineno-3-8" name="__codelineno-3-8" href="#__codelineno-3-8"></a><span class="w">                                 </span><span class="n">min.cells</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0</span><span class="p">,</span><span class="w">                  </span><span class="c1">#Filtering genes that are not detected in less than min.cells</span>
</span><span id="__span-3-9"><a id="__codelineno-3-9" name="__codelineno-3-9" href="#__codelineno-3-9"></a><span class="w">                                 </span><span class="n">min.features</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="w">               </span><span class="c1">#Filtering cells that do not detect at least min.features, here we filter all barcodes that detect no gene</span>
</span></code></pre></div>
<h4 id="exploration-of-the-seuratobject">Exploration of the SeuratObject<a class="headerlink" href="#exploration-of-the-seuratobject" title="Permanent link">&para;</a></h4>
<div class="tabbed-set tabbed-alternate" data-tabs="1:4"><input checked="checked" id="__tabbed_1_1" name="__tabbed_1" type="radio" /><input id="__tabbed_1_2" name="__tabbed_1" type="radio" /><input id="__tabbed_1_3" name="__tabbed_1" type="radio" /><input id="__tabbed_1_4" name="__tabbed_1" type="radio" /><div class="tabbed-labels"><label for="__tabbed_1_1">SeuratObject Presentation</label><label for="__tabbed_1_2">Structure of the SeuratObject</label><label for="__tabbed_1_3">Dimension of expression matrices</label><label for="__tabbed_1_4">Cell Metadata</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<div class="language-r highlight"><pre><span></span><code><span id="__span-4-1"><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a><span class="n">pbmc_small</span><span class="w"> </span><span class="c1">#Small presentation of the Seurat object in R console</span>
</span></code></pre></div>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span>
<span class="normal">2</span>
<span class="normal">3</span></pre></div></td><td class="code"><div><pre><span></span><code><span class="c1">## An object of class Seurat</span>
<span class="c1">## 32738 features across 2700 samples within 1 assay</span>
<span class="c1">## Active assay: RNA (32738 features, 0 variable features)</span>
</code></pre></div></td></tr></table></div>

</div>
<div class="tabbed-block">
<div class="language-r highlight"><pre><span></span><code><span id="__span-5-1"><a id="__codelineno-5-1" name="__codelineno-5-1" href="#__codelineno-5-1"></a><span class="c1">## Discovery of SeuratObject</span>
</span><span id="__span-5-2"><a id="__codelineno-5-2" name="__codelineno-5-2" href="#__codelineno-5-2"></a><span class="nf">str</span><span class="p">(</span><span class="n">pbmc_small</span><span class="p">)</span>
</span></code></pre></div>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal"> 1</span>
<span class="normal"> 2</span>
<span class="normal"> 3</span>
<span class="normal"> 4</span>
<span class="normal"> 5</span>
<span class="normal"> 6</span>
<span class="normal"> 7</span>
<span class="normal"> 8</span>
<span class="normal"> 9</span>
<span class="normal">10</span>
<span class="normal">11</span>
<span class="normal">12</span>
<span class="normal">13</span>
<span class="normal">14</span>
<span class="normal">15</span>
<span class="normal">16</span>
<span class="normal">17</span>
<span class="normal">18</span>
<span class="normal">19</span>
<span class="normal">20</span>
<span class="normal">21</span>
<span class="normal">22</span>
<span class="normal">23</span>
<span class="normal">24</span>
<span class="normal">25</span>
<span class="normal">26</span>
<span class="normal">27</span>
<span class="normal">28</span>
<span class="normal">29</span>
<span class="normal">30</span>
<span class="normal">31</span>
<span class="normal">32</span>
<span class="normal">33</span>
<span class="normal">34</span>
<span class="normal">35</span>
<span class="normal">36</span>
<span class="normal">37</span>
<span class="normal">38</span>
<span class="normal">39</span>
<span class="normal">40</span>
<span class="normal">41</span>
<span class="normal">42</span>
<span class="normal">43</span>
<span class="normal">44</span>
<span class="normal">45</span>
<span class="normal">46</span>
<span class="normal">47</span>
<span class="normal">48</span>
<span class="normal">49</span>
<span class="normal">50</span>
<span class="normal">51</span>
<span class="normal">52</span>
<span class="normal">53</span>
<span class="normal">54</span></pre></div></td><td class="code"><div><pre><span></span><code><span class="c1">## Formal class &#39;Seurat&#39; [package &quot;SeuratObject&quot;] with 13 slots</span>
<span class="c1">##   ..@ assays      :List of 1</span>
<span class="c1">##   .. ..$ RNA:Formal class &#39;Assay5&#39; [package &quot;SeuratObject&quot;] with 8 slots</span>
<span class="c1">##   .. .. .. ..@ layers    :List of 1</span>
<span class="c1">##   .. .. .. .. ..$ counts:Formal class &#39;dgCMatrix&#39; [package &quot;Matrix&quot;] with 6 slots</span>
<span class="c1">##   .. .. .. .. .. .. ..@ i       : int [1:2286884] 70 166 178 326 363 410 412 492 494 495 ...</span>
<span class="c1">##   .. .. .. .. .. .. ..@ p       : int [1:2701] 0 781 2133 3264 4224 4746 5528 6311 7101 7634 ...</span>
<span class="c1">##   .. .. .. .. .. .. ..@ Dim     : int [1:2] 32738 2700</span>
<span class="c1">##   .. .. .. .. .. .. ..@ Dimnames:List of 2</span>
<span class="c1">##   .. .. .. .. .. .. .. ..$ : NULL</span>
<span class="c1">##   .. .. .. .. .. .. .. ..$ : NULL</span>
<span class="c1">##   .. .. .. .. .. .. ..@ x       : num [1:2286884] 1 1 2 1 1 1 1 41 1 1 ...</span>
<span class="c1">##   .. .. .. .. .. .. ..@ factors : list()</span>
<span class="c1">##   .. .. .. ..@ cells     :Formal class &#39;LogMap&#39; [package &quot;SeuratObject&quot;] with 1 slot</span>
<span class="c1">##   .. .. .. .. .. ..@ .Data: logi [1:2700, 1] TRUE TRUE TRUE TRUE TRUE TRUE ...</span>
<span class="c1">##   .. .. .. .. .. .. ..- attr(*, &quot;dimnames&quot;)=List of 2</span>
<span class="c1">##   .. .. .. .. .. .. .. ..$ : chr [1:2700] &quot;AAACATACAACCAC-1&quot; &quot;AAACATTGAGCTAC-1&quot; &quot;AAACATTGATCAGC-1&quot; &quot;AAACCGTGCTTCCG-1&quot; ...</span>
<span class="c1">##   .. .. .. .. .. .. .. ..$ : chr &quot;counts&quot;</span>
<span class="c1">##   .. .. .. .. .. ..$ dim     : int [1:2] 2700 1</span>
<span class="c1">##   .. .. .. .. .. ..$ dimnames:List of 2</span>
<span class="c1">##   .. .. .. .. .. .. ..$ : chr [1:2700] &quot;AAACATACAACCAC-1&quot; &quot;AAACATTGAGCTAC-1&quot; &quot;AAACATTGATCAGC-1&quot; &quot;AAACCGTGCTTCCG-1&quot; ...</span>
<span class="c1">##   .. .. .. .. .. .. ..$ : chr &quot;counts&quot;</span>
<span class="c1">##   .. .. .. ..@ features  :Formal class &#39;LogMap&#39; [package &quot;SeuratObject&quot;] with 1 slot</span>
<span class="c1">##   .. .. .. .. .. ..@ .Data: logi [1:32738, 1] TRUE TRUE TRUE TRUE TRUE TRUE ...</span>
<span class="c1">##   .. .. .. .. .. .. ..- attr(*, &quot;dimnames&quot;)=List of 2</span>
<span class="c1">##   .. .. .. .. .. .. .. ..$ : chr [1:32738] &quot;ENSG00000243485&quot; &quot;ENSG00000237613&quot; &quot;ENSG00000186092&quot; &quot;ENSG00000238009&quot; ...</span>
<span class="c1">##   .. .. .. .. .. .. .. ..$ : chr &quot;counts&quot;</span>
<span class="c1">##   .. .. .. .. .. ..$ dim     : int [1:2] 32738 1</span>
<span class="c1">##   .. .. .. .. .. ..$ dimnames:List of 2</span>
<span class="c1">##   .. .. .. .. .. .. ..$ : chr [1:32738] &quot;ENSG00000243485&quot; &quot;ENSG00000237613&quot; &quot;ENSG00000186092&quot; &quot;ENSG00000238009&quot; ...</span>
<span class="c1">##   .. .. .. .. .. .. ..$ : chr &quot;counts&quot;</span>
<span class="c1">##   .. .. .. ..@ default   : int 1</span>
<span class="c1">##   .. .. .. ..@ assay.orig: chr(0) </span>
<span class="c1">##   .. .. .. ..@ meta.data :&#39;data.frame&#39;:  32738 obs. of  0 variables</span>
<span class="c1">##   .. .. .. ..@ misc      :List of 1</span>
<span class="c1">##   .. .. .. .. ..$ calcN: logi TRUE</span>
<span class="c1">##   .. .. .. ..@ key       : chr &quot;rna_&quot;</span>
<span class="c1">##   ..@ meta.data   :&#39;data.frame&#39;: 2700 obs. of  3 variables:</span>
<span class="c1">##   .. ..$ orig.ident  : Factor w/ 1 level &quot;PBMC analysis&quot;: 1 1 1 1 1 1 1 1 1 1 ...</span>
<span class="c1">##   .. ..$ nCount_RNA  : num [1:2700] 2421 4903 3149 2639 981 ...</span>
<span class="c1">##   .. ..$ nFeature_RNA: int [1:2700] 781 1352 1131 960 522 782 783 790 533 550 ...</span>
<span class="c1">##   ..@ active.assay: chr &quot;RNA&quot;</span>
<span class="c1">##   ..@ active.ident: Factor w/ 1 level &quot;PBMC analysis&quot;: 1 1 1 1 1 1 1 1 1 1 ...</span>
<span class="c1">##   .. ..- attr(*, &quot;names&quot;)= chr [1:2700] &quot;AAACATACAACCAC-1&quot; &quot;AAACATTGAGCTAC-1&quot; &quot;AAACATTGATCAGC-1&quot; &quot;AAACCGTGCTTCCG-1&quot; ...</span>
<span class="c1">##   ..@ graphs      : list()</span>
<span class="c1">##   ..@ neighbors   : list()</span>
<span class="c1">##   ..@ reductions  : list()</span>
<span class="c1">##   ..@ images      : list()</span>
<span class="c1">##   ..@ project.name: chr &quot;PBMC analysis&quot;</span>
<span class="c1">##   ..@ misc        : list()</span>
<span class="c1">##   ..@ version     :Classes &#39;package_version&#39;, &#39;numeric_version&#39;  hidden list of 1</span>
<span class="c1">##   .. ..$ : int [1:3] 5 0 1</span>
<span class="c1">##   ..@ commands    : list()</span>
<span class="c1">##   ..@ tools       : list()</span>
</code></pre></div></td></tr></table></div>

</div>
<div class="tabbed-block">
<div class="language-r highlight"><pre><span></span><code><span id="__span-6-1"><a id="__codelineno-6-1" name="__codelineno-6-1" href="#__codelineno-6-1"></a><span class="nf">dim</span><span class="p">(</span><span class="n">pbmc_small</span><span class="p">[[</span><span class="s">&quot;RNA&quot;</span><span class="p">]])</span>
</span></code></pre></div>
<div class="codehilite"><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span class="normal">1</span></pre></div></td><td class="code"><div><pre><span></span><code>## [1] 32738  2700
</code></pre></div></td></tr></table></div>

</div>
<div class="tabbed-block">
<div class="language-r highlight"><pre><span></span><code><span id="__span-7-1"><a id="__codelineno-7-1" name="__codelineno-7-1" href="#__codelineno-7-1"></a><span class="nf">kable</span><span class="p">(</span><span class="nf">head</span><span class="p">(</span><span class="n">pbmc_small</span><span class="o">@</span><span class="n">meta.data</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;simple&quot;</span><span class="p">)</span><span class="w"> </span><span class="c1">#Preview of the cell metadata</span>
</span></code></pre></div>
<table>
<thead>
<tr>
<th></th>
<th style="text-align: left;">orig.ident</th>
<th style="text-align: right;">nCount_RNA</th>
<th style="text-align: right;">nFeature_RNA</th>
</tr>
</thead>
<tbody>
<tr>
<td>AAACATACAACCAC-1</td>
<td style="text-align: left;">PBMC analysis</td>
<td style="text-align: right;">2421</td>
<td style="text-align: right;">781</td>
</tr>
<tr>
<td>AAACATTGAGCTAC-1</td>
<td style="text-align: left;">PBMC analysis</td>
<td style="text-align: right;">4903</td>
<td style="text-align: right;">1352</td>
</tr>
<tr>
<td>AAACATTGATCAGC-1</td>
<td style="text-align: left;">PBMC analysis</td>
<td style="text-align: right;">3149</td>
<td style="text-align: right;">1131</td>
</tr>
<tr>
<td>AAACCGTGCTTCCG-1</td>
<td style="text-align: left;">PBMC analysis</td>
<td style="text-align: right;">2639</td>
<td style="text-align: right;">960</td>
</tr>
<tr>
<td>AAACCGTGTATGCG-1</td>
<td style="text-align: left;">PBMC analysis</td>
<td style="text-align: right;">981</td>
<td style="text-align: right;">522</td>
</tr>
<tr>
<td>AAACGCACTGGTAC-1</td>
<td style="text-align: left;">PBMC analysis</td>
<td style="text-align: right;">2164</td>
<td style="text-align: right;">782</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<p>We can observe several slots via the <code>str</code> command:  </p>
<ul>
<li><code>assays</code>: general slot that will include the different information
    of each study. Its a list where each element is linked to a
    expression data type. For instance in the element <code>RNA</code> you retrieve
    all information link to you scRNAseq. But for example, if you have a
    more complicated study you can also have <code>ADT</code> or <code>peaks</code> if you
    perform a CITEseq, scATACseq or Multiome. Here we have a simple
    scRNAseq, so by default <code>assays</code> will be composed of only the
    element <code>RNA</code>. Its class is <code>Assay5</code> you can have a great
    description in the documentation of the <code>SeuratObject</code> package. In
    you console <code>?Assay5</code> to read it.</li>
<li><code>meta.data</code> : gathers all the information about the cells. At the
    beginning Seurat will calculate the size of the library (nCount_RNA,
    or the total number of UMI) and the number of detected genes
    (nFeatures_RNA) for each cell. If a dataframe is given in the
    <code>meta.data</code> parameter of the <code>CreateSeuratObject</code> function, its
    columns will be added after those calculated by Seurat.</li>
<li><code>active.assay</code> : study used by default</li>
<li><code>active.ident</code> : default cell identity, here the name of the given
    project, also stored under the column <code>orig.ident</code> in the metadata</li>
</ul>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Navigation in the different slots is done via <code>@</code> or <code>$</code>. Each main slot is
accessible via the <code>@</code>, <em>i.e.</em> <code>object@main slot</code> to go further in the
slots tree, most often complex objects are accessible with a <code>@</code> (dgCMatrix,
dataframe) and lists, vectors are accessible via <code>$</code>. If in doubt, you can
refer to the result of the <code>str</code> command and use the character in front of
each slot name. If you want to look to the expression matrix you can go :</p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-8-1"><a id="__codelineno-8-1" name="__codelineno-8-1" href="#__codelineno-8-1"></a><span class="c1"># Retrieve data in an expression matrix RNA counts matrix</span>
</span><span id="__span-8-2"><a id="__codelineno-8-2" name="__codelineno-8-2" href="#__codelineno-8-2"></a><span class="n">pbmc_small</span><span class="p">[[</span><span class="s">&quot;RNA&quot;</span><span class="p">]]</span><span class="o">$</span><span class="n">counts</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">10</span><span class="p">,</span><span class="m">1</span><span class="o">:</span><span class="m">5</span><span class="p">]</span>
</span></code></pre></div>
</div>












                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var tab,labels=set.querySelector(".tabbed-labels");for(tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
    
      
      <nav class="md-footer__inner md-grid" aria-label="Footer" >
        
          
          <a href="../intro_seurat/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Initialization of R analysis">
            <div class="md-footer__button md-icon">
              
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
            </div>
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Previous
              </span>
              <div class="md-ellipsis">
                Initialization of R analysis
              </div>
            </div>
          </a>
        
        
          
          <a href="../preprocessing/" class="md-footer__link md-footer__link--next" aria-label="Next: Preprocessing">
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Next
              </span>
              <div class="md-ellipsis">
                Preprocessing
              </div>
            </div>
            <div class="md-footer__button md-icon">
              
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg>
            </div>
          </a>
        
      </nav>
    
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "../..", "features": ["header.autohide", "navigation.instant", "navigation.tracking", "navigation.tabs", "navigation.tabs.sticky", "navigation.top", "navigation.prune", "navigation.footer", "content.tabs.link", "content.code.copy", "content.action.view", "content.action.edit"], "search": "../../assets/javascripts/workers/search.b8dbb3d2.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.1e8ae164.min.js"></script>
      
        <script src="../../javascripts/mathjax.js"></script>
      
        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
      
        <script src="../../spoiler_files/spoiler.js"></script>
      
    
  </body>
</html>