
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://artbio.github.io/startbio/bulk_RNAseq-IOC/28_manipulation_DEseq2_data/">
      
      
        <link rel="prev" href="../27_1_limma/">
      
      
        <link rel="next" href="../28_1_manipulation_edgeR_data/">
      
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.5.3, mkdocs-material-9.5.17">
    
    
      
        <title>DESeq2 - STARTbio</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.bcfcd587.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.06af60db.min.css">
      
      
  
  
    
    
  
  
  <style>:root{--md-admonition-icon--danger:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 12"><path d="M4.855.708c.5-.896 1.79-.896 2.29 0l4.675 8.351a1.312 1.312 0 0 1-1.146 1.954H1.33A1.313 1.313 0 0 1 .183 9.058ZM7 7V3H5v4Zm-1 3a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z"/></svg>');}</style>



    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Noto+Sans+Display:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Noto Sans Display";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../spoiler_files/spoiler.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="black" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#manipulation-of-deseq2-data-for-visualisation-and-comparisons" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="STARTbio" class="md-header__button md-logo" aria-label="STARTbio" data-md-component="logo">
      
  <img src="../../assets/startbio_icon.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            STARTbio
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              DESeq2
            
          </span>
        </div>
      </div>
    </div>
    
      
    
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/ARTbio/startbio" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    ARTbio/startbio
  </div>
</a>
      </div>
    
  </nav>
  
    
      
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../.." class="md-tabs__link">
        
  
    
  
  Home

      </a>
    </li>
  

      
        
  
  
    
  
  
    
    
      
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="../00_IOC_RNAseq_program/" class="md-tabs__link">
          
  
  Interactive Online Companionship (IOC)

        </a>
      </li>
    
  

    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../ten2/Regex_01/" class="md-tabs__link">
          
  
  1Omin2

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../AnalyseGenomes_2025/outline/" class="md-tabs__link">
          
  
  Galaxy Trainings

        </a>
      </li>
    
  

    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../bash_snippets/" class="md-tabs__link">
          
  
  Code Snippets

        </a>
      </li>
    
  

      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../../links/" class="md-tabs__link">
        
  
    
  
  Links

      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
    
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="STARTbio" class="md-nav__button md-logo" aria-label="STARTbio" data-md-component="logo">
      
  <img src="../../assets/startbio_icon.png" alt="logo">

    </a>
    STARTbio
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/ARTbio/startbio" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    ARTbio/startbio
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Home
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
      
        
        
      
      
    
    
      
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" checked>
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="">
            
  
  <span class="md-ellipsis">
    Interactive Online Companionship (IOC)
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            Interactive Online Companionship (IOC)
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
    
  
  
  
    
    
    
      
      
    
    
      
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_1" checked>
        
          
          <label class="md-nav__link" for="__nav_2_1" id="__nav_2_1_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    IOC on bulk RNAseq analysis
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_1_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_2_1">
            <span class="md-nav__icon md-icon"></span>
            IOC on bulk RNAseq analysis
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../00_IOC_RNAseq_program/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Provisional Program / Schedule
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../01_IOC_RNAseq_week_00/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Week 0
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../02_exercices_week_00_review/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Week 1
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../13_exercices_week_01_review/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Week 2
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../19_exercices_week_02_review/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Week 3
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

              
            
              
                
  
  
    
  
  
  
    
    
    
      
      
    
    
      
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_1_6" checked>
        
          
          <label class="md-nav__link" for="__nav_2_1_6" id="__nav_2_1_6_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Week 4
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_2_1_6_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_2_1_6">
            <span class="md-nav__icon md-icon"></span>
            Week 4
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../24_exercices_week_03_review/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Review on week-3 work
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../25_DE_intro/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Differential Gene Expression Analysis
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    
    
    
      
      
    
    
      
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_1_6_3" checked>
        
          
          <label class="md-nav__link" for="__nav_2_1_6_3" id="__nav_2_1_6_3_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Week 4 exercices
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_2_1_6_3_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_2_1_6_3">
            <span class="md-nav__icon md-icon"></span>
            Week 4 exercices
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../26_deseq2/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    DESeq2
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../27_edgeR/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    edgeR
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../27_1_limma/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    limma
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    
    
    
      
      
    
    
      
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_1_6_3_4" checked>
        
          
          <label class="md-nav__link" for="__nav_2_1_6_3_4" id="__nav_2_1_6_3_4_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    DE_manipulations
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="5" aria-labelledby="__nav_2_1_6_3_4_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_2_1_6_3_4">
            <span class="md-nav__icon md-icon"></span>
            DE_manipulations
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    DESeq2
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    DESeq2
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#extract-the-most-differentially-expressed-genes-prjna630433-deseq2" class="md-nav__link">
    <span class="md-ellipsis">
      Extract the most differentially expressed genes (PRJNA630433 / DESeq2)
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#generate-top-lists-of-de-genes" class="md-nav__link">
    <span class="md-ellipsis">
      Generate top lists of DE genes
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Generate top lists of DE genes">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#select-genes-with-log2fc-2-and-p-adj-001-with-filter-data-on-any-column-using-simple-expressions" class="md-nav__link">
    <span class="md-ellipsis">
      Select genes with |log2FC &gt; 2| and p-adj &lt; 0.01 with Filter data on any column using simple expressions
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#compute-a-boolean-value-by-row" class="md-nav__link">
    <span class="md-ellipsis">
      Compute a boolean value by row
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#transform-true-and-false-values-to-up-and-down-respectively" class="md-nav__link">
    <span class="md-ellipsis">
      Transform True and False values to up and down, respectively
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#split-the-lists-in-up-and-down-regulated-lists" class="md-nav__link">
    <span class="md-ellipsis">
      Split the lists in up and down regulated lists
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#plotting-an-heatmap-of-the-most-significantly-de-regulated-genes" class="md-nav__link">
    <span class="md-ellipsis">
      Plotting an heatmap of the most significantly de-regulated genes
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Plotting an heatmap of the most significantly de-regulated genes">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#use-advanced-cut-to-select-the-list-of-deregulated-genes-in-all-three-comparisons" class="md-nav__link">
    <span class="md-ellipsis">
      Use Advanced cut to select the list of deregulated genes in all three comparisons
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#next-we-concatenate-the-three-datasets-of-the-previous-collection-in-a-single-dataset" class="md-nav__link">
    <span class="md-ellipsis">
      Next we concatenate the three datasets of the previous collection in a single dataset
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#next-we-extract-uniques-gene-names-from-the-pooled-top-genes-dataset" class="md-nav__link">
    <span class="md-ellipsis">
      Next we extract Uniques gene names from the Pooled top genes dataset
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#add-a-header-the-list-of-unique-gene-names-associated-we-significant-de-in-any-of-the-comparisons" class="md-nav__link">
    <span class="md-ellipsis">
      Add a header the list of unique gene names associated we significant DE in any of the comparisons
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#intersection-join-operation-between-the-list-of-unique-gene-name-associated-with-de-and-the-rlog-normalized-counts-file" class="md-nav__link">
    <span class="md-ellipsis">
      Intersection (join operation) between the list of unique gene name associated with DE and the rLog-Normalized counts file.
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#plot-a-heatmap-of-the-rlog-normalized-counts-of-de-genes-in-all-three-conditions" class="md-nav__link">
    <span class="md-ellipsis">
      Plot a heatmap of the rLog-Normalized counts of DE genes in all three conditions
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../28_1_manipulation_edgeR_data/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    edgeR
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../28_2_manipulation_limma_data/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    limma
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../29_0_comparisons/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    DEseq/edgeR/limma comparison
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../29_volcano/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Volcano plots
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../30_exercices_week_04_review/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Week 5
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../33_exercices_week_05_review/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Week 6
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../40_exercices_week_06_review/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Week 7
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../50_exercices_week_07_review/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Week 8
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../links/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    External Materials
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    
  
  
    <a href="../../R-IOC/00_IOC_R_program/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    IOC on R
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

  

      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    
  
  
    <a href="../../scRNAseq_basics/program_sc/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    IOC on Single-cell RNAseq analysis using Seurat in R
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

  

      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../../deepseq/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Introduction to Deep-seq Analysis
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../../ten2/Regex_01/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    1Omin2
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    
  
  
    <a href="../../AnalyseGenomes_2025/outline/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Galaxy Trainings
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

  

      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../../bash_snippets/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Code Snippets
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../links/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Links
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#extract-the-most-differentially-expressed-genes-prjna630433-deseq2" class="md-nav__link">
    <span class="md-ellipsis">
      Extract the most differentially expressed genes (PRJNA630433 / DESeq2)
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#generate-top-lists-of-de-genes" class="md-nav__link">
    <span class="md-ellipsis">
      Generate top lists of DE genes
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Generate top lists of DE genes">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#select-genes-with-log2fc-2-and-p-adj-001-with-filter-data-on-any-column-using-simple-expressions" class="md-nav__link">
    <span class="md-ellipsis">
      Select genes with |log2FC &gt; 2| and p-adj &lt; 0.01 with Filter data on any column using simple expressions
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#compute-a-boolean-value-by-row" class="md-nav__link">
    <span class="md-ellipsis">
      Compute a boolean value by row
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#transform-true-and-false-values-to-up-and-down-respectively" class="md-nav__link">
    <span class="md-ellipsis">
      Transform True and False values to up and down, respectively
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#split-the-lists-in-up-and-down-regulated-lists" class="md-nav__link">
    <span class="md-ellipsis">
      Split the lists in up and down regulated lists
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#plotting-an-heatmap-of-the-most-significantly-de-regulated-genes" class="md-nav__link">
    <span class="md-ellipsis">
      Plotting an heatmap of the most significantly de-regulated genes
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Plotting an heatmap of the most significantly de-regulated genes">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#use-advanced-cut-to-select-the-list-of-deregulated-genes-in-all-three-comparisons" class="md-nav__link">
    <span class="md-ellipsis">
      Use Advanced cut to select the list of deregulated genes in all three comparisons
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#next-we-concatenate-the-three-datasets-of-the-previous-collection-in-a-single-dataset" class="md-nav__link">
    <span class="md-ellipsis">
      Next we concatenate the three datasets of the previous collection in a single dataset
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#next-we-extract-uniques-gene-names-from-the-pooled-top-genes-dataset" class="md-nav__link">
    <span class="md-ellipsis">
      Next we extract Uniques gene names from the Pooled top genes dataset
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#add-a-header-the-list-of-unique-gene-names-associated-we-significant-de-in-any-of-the-comparisons" class="md-nav__link">
    <span class="md-ellipsis">
      Add a header the list of unique gene names associated we significant DE in any of the comparisons
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#intersection-join-operation-between-the-list-of-unique-gene-name-associated-with-de-and-the-rlog-normalized-counts-file" class="md-nav__link">
    <span class="md-ellipsis">
      Intersection (join operation) between the list of unique gene name associated with DE and the rLog-Normalized counts file.
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#plot-a-heatmap-of-the-rlog-normalized-counts-of-de-genes-in-all-three-conditions" class="md-nav__link">
    <span class="md-ellipsis">
      Plot a heatmap of the rLog-Normalized counts of DE genes in all three conditions
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
    <a href="https://github.com/ARTbio/startbio/edit/master/docs/bulk_RNAseq-IOC/28_manipulation_DEseq2_data.md" title="Edit this page" class="md-content__button md-icon">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M10 20H6V4h7v5h5v3.1l2-2V8l-6-6H6c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h4v-2m10.2-7c.1 0 .3.1.4.2l1.3 1.3c.2.2.2.6 0 .8l-1 1-2.1-2.1 1-1c.1-.1.2-.2.4-.2m0 3.9L14.1 23H12v-2.1l6.1-6.1 2.1 2.1Z"/></svg>
    </a>
  
  
    
      
    
    <a href="https://github.com/ARTbio/startbio/raw/master/docs/bulk_RNAseq-IOC/28_manipulation_DEseq2_data.md" title="View source of this page" class="md-content__button md-icon">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 18c.56 0 1 .44 1 1s-.44 1-1 1-1-.44-1-1 .44-1 1-1m0-3c-2.73 0-5.06 1.66-6 4 .94 2.34 3.27 4 6 4s5.06-1.66 6-4c-.94-2.34-3.27-4-6-4m0 6.5a2.5 2.5 0 0 1-2.5-2.5 2.5 2.5 0 0 1 2.5-2.5 2.5 2.5 0 0 1 2.5 2.5 2.5 2.5 0 0 1-2.5 2.5M9.27 20H6V4h7v5h5v4.07c.7.08 1.36.25 2 .49V8l-6-6H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h4.5a8.15 8.15 0 0 1-1.23-2Z"/></svg>
    </a>
  


<h1 id="manipulation-of-deseq2-data-for-visualisation-and-comparisons">Manipulation of DESeq2 data for visualisation and comparisons<a class="headerlink" href="#manipulation-of-deseq2-data-for-visualisation-and-comparisons" title="Permanent link">&para;</a></h1>
<p>Now we would like to extract the most differentially expressed genes in the various
conditions, and then visualize them using an heatmap of the normalized counts for each
sample.</p>
<p>We will proceed in several steps:</p>
<ul class="task-list">
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> For each package, extract the normalized counts of genes for each sample (all three
  packages, DESeq2, edgeR and limma, provide this functionality.</li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> For each package, extract the most differentially expressed genes at a given log2FC
  threshold (let's say 2, corresponding to a 4x or &frac14;x fold time expression), and at a
  given p-adjusted value (let's say p-adj &lt; 0.01). We will keep these gene lists apart to
  build latter a venn diagram for comparison of the three tools.</li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> Plot heatmaps of normalized counts</li>
</ul>
<h2 id="extract-the-most-differentially-expressed-genes-prjna630433-deseq2">Extract the most differentially expressed genes (PRJNA630433 / DESeq2)<a class="headerlink" href="#extract-the-most-differentially-expressed-genes-prjna630433-deseq2" title="Permanent link">&para;</a></h2>
<p>Basically, we navigate in the DESeq history of the PRJNA630433 use-case and we repeat a
DESeq2 run, asking in addition for a <strong>rLog-Normalized</strong> counts output.</p>
<details class="info">
<summary><img align="absbottom" alt="" src="../images/tool_small.png" width="25" /> <code>DESeq2</code> settings</summary>
<p>Basically, the same as before, except that we ask for a Normalized counts file</p>
<ul>
<li>
<p>how</p>
<p>&rarr; Select datasets per levels</p>
</li>
<li>
<p>1: Factor</p>
<p>&rarr; Tissue</p>
</li>
<li>
<p>1: Factor level</p>
<p>Note that there will be three factor levels in this analysis: Dc, Mo and Oc.</p>
<p>&rarr; Oc</p>
</li>
<li>
<p>Counts file(s)</p>
<p>&rarr; select the data collection icon, then <code>15: Oc FeatureCounts counts</code></p>
</li>
<li>
<p>2: Factor level</p>
<p>&rarr; Mo</p>
</li>
<li>
<p>Counts file(s)</p>
<p>&rarr; select the data collection icon, then <code>10: Mo FeatureCounts counts</code></p>
</li>
<li>
<p>3: Factor level (you must click on <img alt="➕" class="twemoji" src="https://cdn.jsdelivr.net/gh/jdecked/twemoji@15.0.3/assets/svg/2795.svg" title=":heavy_plus_sign:" /> <code>Insert Factor level</code>)</p>
<p>&rarr; Dc</p>
</li>
<li>
<p>Counts file(s)</p>
<p>&rarr; select the data collection icon, then <code>5: Mo FeatureCounts counts</code></p>
</li>
<li>
<p>(Optional) provide a tabular file with additional batch factors to include in the model.</p>
<p>&rarr; Leave to <code>Nothing selected</code></p>
</li>
<li>
<p>Files have header?</p>
<p>&rarr; Yes</p>
</li>
<li>
<p>Choice of Input data</p>
<p>&rarr; Count data</p>
</li>
<li>
<p>Advanced options</p>
<p>&rarr; No, leave folded</p>
</li>
<li>
<p>Output options</p>
<p>&rarr; <mark>This time, check the <code>Output rLog normalized table</code> box !</mark></p>
<p>&rarr; Unfold and check <code>Output all levels vs all levels of primary factor (use when
you have &gt;2 levels for primary factor)</code> in addition to the already checked
<code>Generate plots for visualizing the analysis results</code></p>
<p>&rarr; Leave <code>Alpha value for MA-plot</code> to 0,1: note that this option is used for
plots and does not impact DESeq2 results</p>
</li>
<li>
<p><code>Run Tool</code></p>
</li>
</ul>
</details>
<p><img alt="⚠" class="twemoji" src="https://cdn.jsdelivr.net/gh/jdecked/twemoji@15.0.3/assets/svg/26a0.svg" title=":warning:" /> This time you can trash the DESeq2 plots and result files which we have already
generated.</p>
<p><img alt="⚠" class="twemoji" src="https://cdn.jsdelivr.net/gh/jdecked/twemoji@15.0.3/assets/svg/26a0.svg" title=":warning:" /> Keep this output for latter, will use it for a clustered heatmap</p>
<h2 id="generate-top-lists-of-de-genes">Generate top lists of DE genes<a class="headerlink" href="#generate-top-lists-of-de-genes" title="Permanent link">&para;</a></h2>
<p>We will do that with the help of the tool <code>Filter data on any column using simple
expressions</code>. We will also use 3 other tools <code>Compute on rows</code>, <code>Column Regex Find And
Replace</code> and <code>Filter data on any column using simple expressions</code></p>
<h3 id="select-genes-with-log2fc-2-and-p-adj-001-with-filter-data-on-any-column-using-simple-expressions">Select genes with |log2FC &gt; 2| and p-adj &lt; 0.01 with <img align="absbottom" alt="" src="../images/tool_small.png" width="30" /><code>Filter data on any column using simple expressions</code><a class="headerlink" href="#select-genes-with-log2fc-2-and-p-adj-001-with-filter-data-on-any-column-using-simple-expressions" title="Permanent link">&para;</a></h3>
<div class="admonition info">
<p class="admonition-title"><img align="absbottom" alt="" src="../images/tool_small.png" width="25" /> <code>Filter data on any column...</code> settings</p>
<ul>
<li>
<p>Filter</p>
<p>&rarr; DESeq2 Results Tables</p>
</li>
<li>
<p>With following condition</p>
<p>&rarr; abs(c3) &gt; 2 and c7 &lt; 0.01</p>
</li>
<li>
<p>Number of header lines to skip</p>
<p>&rarr; <code>1</code> (these tables have an added header !)</p>
</li>
<li>
<p>Click the <code>Run Tool</code> button</p>
</li>
</ul>
</div>
<p><img alt="⚠" class="twemoji" src="https://cdn.jsdelivr.net/gh/jdecked/twemoji@15.0.3/assets/svg/26a0.svg" title=":warning:" /> Rename the "filter on..." collection to <code>Top gene lists</code></p>
<h3 id="compute-a-boolean-value-by-row">Compute a boolean value by row<a class="headerlink" href="#compute-a-boolean-value-by-row" title="Permanent link">&para;</a></h3>
<p>This is to determine whether genes in the lists are up or down-regulated</p>
<div class="admonition info">
<p class="admonition-title"><img align="absbottom" alt="" src="../images/tool_small.png" width="25" /> <code>Compute on rows</code> settings</p>
<ul>
<li>
<p>Input file</p>
<p>&rarr; <code>Top gene lists</code> (<img alt="⚠" class="twemoji" src="https://cdn.jsdelivr.net/gh/jdecked/twemoji@15.0.3/assets/svg/26a0.svg" title=":warning:" /> collection !)</p>
</li>
<li>
<p>Input has a header line with column names?</p>
<p>&rarr; <code>Yes</code></p>
</li>
<li>
<p>1: Expressions</p>
</li>
<li>
<p>Add expression</p>
<p>&rarr; <code>c3 &gt; 0</code></p>
</li>
<li>
<p>Mode of the operation</p>
<p>&rarr; <code>Append</code></p>
</li>
<li>
<p>The new column name</p>
<p>&rarr; <code>Regulation</code></p>
</li>
<li>
<p>Avoid scientific notation in any newly computed columns</p>
<p>&rarr; <code>No</code></p>
</li>
<li>
<p>Click the <code>Run Tool</code> button</p>
</li>
</ul>
</div>
<p><img alt="⚠" class="twemoji" src="https://cdn.jsdelivr.net/gh/jdecked/twemoji@15.0.3/assets/svg/26a0.svg" title=":warning:" /> Look at the effect of evaluating the expression <code>c3 &gt; 0</code> in the new column
<code>expression</code> in the output datasets.</p>
<h3 id="transform-true-and-false-values-to-up-and-down-respectively">Transform <code>True</code> and <code>False</code> values to <code>up</code> and <code>down</code>, respectively<a class="headerlink" href="#transform-true-and-false-values-to-up-and-down-respectively" title="Permanent link">&para;</a></h3>
<div class="admonition info">
<p class="admonition-title"><img align="absbottom" alt="" src="../images/tool_small.png" width="25" /> <code>Column Regex Find And Replace</code> settings</p>
<ul>
<li>
<p>Select cells from</p>
<p>&rarr; <code>Compute on collection 36 (or so)</code></p>
</li>
<li>
<p>using column</p>
<p>&rarr; <code>8</code></p>
</li>
<li>
<p>Check</p>
<p>&rarr; click the button <img alt="➕" class="twemoji" src="https://cdn.jsdelivr.net/gh/jdecked/twemoji@15.0.3/assets/svg/2795.svg" title=":heavy_plus_sign:" /><code>Insert Check</code></p>
</li>
<li>
<p>Find Regex</p>
<p>&rarr; <code>False</code></p>
</li>
<li>
<p>Replacement</p>
<p>&rarr; <code>down</code></p>
</li>
<li>
<p>Check</p>
<p>&rarr; click another time the button <img alt="➕" class="twemoji" src="https://cdn.jsdelivr.net/gh/jdecked/twemoji@15.0.3/assets/svg/2795.svg" title=":heavy_plus_sign:" /><code>Insert Check</code></p>
</li>
<li>
<p>Find Regex</p>
<p>&rarr; <code>True</code></p>
</li>
<li>
<p>Replacement</p>
<p>&rarr; <code>up</code></p>
</li>
<li>
<p>Click the <code>Run Tool</code> button</p>
</li>
</ul>
</div>
<p><img alt="⚠" class="twemoji" src="https://cdn.jsdelivr.net/gh/jdecked/twemoji@15.0.3/assets/svg/26a0.svg" title=":warning:" /> rename the collection <code>Column Regex Find And Replace on collection 40</code> with
<code>top gene lists - oriented</code></p>
<h3 id="split-the-lists-in-up-and-down-regulated-lists">Split the lists in <code>up</code> and <code>down</code> regulated lists<a class="headerlink" href="#split-the-lists-in-up-and-down-regulated-lists" title="Permanent link">&para;</a></h3>
<p>This will be performed through 2 successive runs of the
<img align="absbottom" alt="" src="../images/tool_small.png" width="25" /> tool <code>Select lines that match an
expression</code></p>
<div class="admonition info">
<p class="admonition-title"><img align="absbottom" alt="" src="../images/tool_small.png" width="25" /> <code>Select lines that match an expression</code> settings</p>
<ul>
<li>
<p>Select lines from</p>
<p>&rarr; <code>top gene lists - oriented</code></p>
</li>
<li>
<p>that</p>
<p>&rarr; <code>matching</code></p>
</li>
<li>
<p>the pattern</p>
<p>&rarr; <code>\tup</code> (a tabulation immediately followed by the string <em>up</em>)</p>
</li>
<li>
<p>Keep header line</p>
<p>&rarr; <code>Yes</code></p>
</li>
<li>
<p>Click the <code>Run Tool</code> button</p>
</li>
</ul>
</div>
<p><img alt="⚠" class="twemoji" src="https://cdn.jsdelivr.net/gh/jdecked/twemoji@15.0.3/assets/svg/26a0.svg" title=":warning:" /> Immediately rename the collection <code>Select on collection...</code> to <code>top up-regulated
gene lists</code></p>
<p>Redo exactly the same operation with a single change in the setting of the
<img align="absbottom" alt="" src="../images/tool_small.png" width="25" /> tool <code>Select lines that match an
expression</code></p>
<details class="info">
<summary><img align="absbottom" alt="" src="../images/tool_small.png" width="25" /> <code>Select lines that match an expression</code> settings</summary>
<ul>
<li>
<p>Select lines from</p>
<p>&rarr; <code>top gene lists - oriented</code></p>
</li>
<li>
<p>that</p>
<p>&rarr; <code>matching</code></p>
</li>
<li>
<p>the pattern</p>
<p>&rarr; <code>\tdown</code> (a tabulation immediately followed by the string <em>down</em>)</p>
</li>
<li>
<p>Keep header line</p>
<p>&rarr; <code>Yes</code></p>
</li>
<li>
<p>Click the <code>Run Tool</code> button</p>
</li>
</ul>
</details>
<p><img alt="⚠" class="twemoji" src="https://cdn.jsdelivr.net/gh/jdecked/twemoji@15.0.3/assets/svg/26a0.svg" title=":warning:" /> Rename the collection <code>Select on collection...</code> to <code>top down-regulated
gene lists</code></p>
<p><img alt="⚠" class="twemoji" src="https://cdn.jsdelivr.net/gh/jdecked/twemoji@15.0.3/assets/svg/26a0.svg" title=":warning:" /> keep the last three generated collections for later comparison with edgeR and
limma tools</p>
<h2 id="plotting-an-heatmap-of-the-most-significantly-de-regulated-genes">Plotting an heatmap of the most significantly de-regulated genes<a class="headerlink" href="#plotting-an-heatmap-of-the-most-significantly-de-regulated-genes" title="Permanent link">&para;</a></h2>
<p>For this, we are going to collect and gather all significantly de-regulated genes in any of the
3 conditions, and to intersect (join operation) this list with the rLog normalized count
table precedently generated.</p>
<h3 id="use-advanced-cut-to-select-the-list-of-deregulated-genes-in-all-three-comparisons">Use <img align="absbottom" alt="" src="../images/tool_small.png" width="25" /><code>Advanced cut</code> to select the list of deregulated genes in all three comparisons<a class="headerlink" href="#use-advanced-cut-to-select-the-list-of-deregulated-genes-in-all-three-comparisons" title="Permanent link">&para;</a></h3>
<div class="admonition info">
<p class="admonition-title"><img align="absbottom" alt="" src="../images/tool_small.png" width="25" /> <code>advanced cut</code> settings</p>
<ul>
<li>
<p>File to cut</p>
<p>&rarr; <code>Top gene lists</code> (this is a collection)</p>
</li>
<li>
<p>Operation</p>
<p>&rarr; <code>Keep</code></p>
</li>
<li>
<p>Delimited by</p>
<p>&rarr; <code>Tab</code></p>
</li>
<li>
<p>Cut by</p>
<p>&rarr; <code>fields</code></p>
</li>
<li>
<p>List of Fields</p>
<p>&rarr; <code>Column 1</code></p>
</li>
<li>
<p>First line is a header line</p>
</li>
<li>Click the <code>Run Tool</code> button</li>
</ul>
</div>
<p><img alt="⚠" class="twemoji" src="https://cdn.jsdelivr.net/gh/jdecked/twemoji@15.0.3/assets/svg/26a0.svg" title=":warning:" /> Rename this collection of single column datasets <code>top genes names</code></p>
<h3 id="next-we-concatenate-the-three-datasets-of-the-previous-collection-in-a-single-dataset">Next we concatenate the three datasets of the previous collection in a single dataset<a class="headerlink" href="#next-we-concatenate-the-three-datasets-of-the-previous-collection-in-a-single-dataset" title="Permanent link">&para;</a></h3>
<p>We do that using the <img align="absbottom" alt="" src="../images/tool_small.png" width="25" />
<code>Concatenate multiple datasets tail-to-head while specifying how</code> tool</p>
<div class="admonition info">
<p class="admonition-title"><img align="absbottom" alt="" src="../images/tool_small.png" width="25" /> <code>Concatenate multiple datasets tail-to-head while specifying how</code> settings</p>
<ul>
<li>
<p>What type of data do you wish to concatenate?</p>
<p>&rarr; <code>Single datasets</code></p>
</li>
<li>
<p>Concatenate Datasets</p>
<p>&rarr; <img alt="⚠" class="twemoji" src="https://cdn.jsdelivr.net/gh/jdecked/twemoji@15.0.3/assets/svg/26a0.svg" title=":warning:" /> Click on the collection icon and select <code>top genes names</code></p>
</li>
<li>
<p>Include dataset names?</p>
<p>&rarr; <code>No</code></p>
</li>
<li>
<p>Number of lines to skip at the beginning of each concatenation:</p>
<p>&rarr; <code>1</code></p>
</li>
<li>
<p>Click the <code>Run Tool</code> button</p>
</li>
</ul>
</div>
<p><img alt="⚠" class="twemoji" src="https://cdn.jsdelivr.net/gh/jdecked/twemoji@15.0.3/assets/svg/26a0.svg" title=":warning:" /> Rename the return single dataset as <code>Pooled top genes</code></p>
<h3 id="next-we-extract-uniques-gene-names-from-the-pooled-top-genes-dataset">Next we extract <em>Uniques</em> gene names from the <code>Pooled top genes</code> dataset<a class="headerlink" href="#next-we-extract-uniques-gene-names-from-the-pooled-top-genes-dataset" title="Permanent link">&para;</a></h3>
<p>You probably agree that the same gene may be deregulated in the three pair-wise comparisons
which we have performed with DESeq2.</p>
<p>Thus we need to eliminate the redundancy, using the tool
<img align="absbottom" alt="" src="../images/tool_small.png" width="25" /><code>Unique
occurrences of each record</code>.</p>
<div class="admonition info">
<p class="admonition-title"><img align="absbottom" alt="" src="../images/tool_small.png" width="25" /> <code>Unique occurrences of each record</code> settings</p>
<ul>
<li>
<p>File to scan for unique values</p>
<p>&rarr; <code>Pooled top genes</code></p>
</li>
<li>
<p>Ignore differences in case when comparing</p>
<p>&rarr; <code>No</code></p>
</li>
<li>
<p>Column only contains numeric values</p>
<p>&rarr; <code>No</code></p>
</li>
<li>
<p>Advanced Options</p>
<p>&rarr; Leave as <code>Hide Advanced Options</code></p>
</li>
<li>
<p>Click the <code>Run Tool</code> button</p>
</li>
</ul>
</div>
<h3 id="add-a-header-the-list-of-unique-gene-names-associated-we-significant-de-in-any-of-the-comparisons">Add a header the list of unique gene names associated we significant DE in any of the comparisons<a class="headerlink" href="#add-a-header-the-list-of-unique-gene-names-associated-we-significant-de-in-any-of-the-comparisons" title="Permanent link">&para;</a></h3>
<p>We do this with the tools <code>Add Header</code></p>
<div class="admonition info">
<p class="admonition-title"><img align="absbottom" alt="" src="../images/tool_small.png" width="25" /> <code>Add Header</code> settings</p>
<ul>
<li>
<p>List of Column headers (comma delimited, e.g. C1,C2,...)</p>
<p>&rarr; <code>DESeq_All_DE_genes</code></p>
</li>
<li>
<p>Data File (tab-delimted)</p>
<p>&rarr; <code>Unique on data 1xx...</code></p>
</li>
<li>
<p>Click the <code>Run Tool</code> button</p>
</li>
</ul>
</div>
<p><img alt="⚠" class="twemoji" src="https://cdn.jsdelivr.net/gh/jdecked/twemoji@15.0.3/assets/svg/26a0.svg" title=":warning:" /> Rename the generated dataset <code>DESeq_All_DE_genes</code></p>
<h3 id="intersection-join-operation-between-the-list-of-unique-gene-name-associated-with-de-and-the-rlog-normalized-counts-file">Intersection (join operation) between the list of unique gene name associated with DE and the rLog-Normalized counts file.<a class="headerlink" href="#intersection-join-operation-between-the-list-of-unique-gene-name-associated-with-de-and-the-rlog-normalized-counts-file" title="Permanent link">&para;</a></h3>
<p>This is the moment when we are going to use the <code>rLog-Normalized counts file on data...</code>
and intersect it (join operation) with the list of DE genes in all three condition.</p>
<p>To do this, we are going to use the tool
<img align="absbottom" alt="" src="../images/tool_small.png" width="25" /><code>Join two files</code></p>
<div class="admonition info">
<p class="admonition-title"><img align="absbottom" alt="" src="../images/tool_small.png" width="25" /> <code>Join two files</code> settings</p>
<ul>
<li>
<p>1<sup>st</sup> file</p>
<p>&rarr; <code>rLog-Normalized counts file on data...</code></p>
</li>
<li>
<p>Column to use from 1<sup>st</sup> file</p>
<p>&rarr; <code>1</code></p>
</li>
<li>
<p>2<sup>nd</sup> File</p>
<p>&rarr; <code>DESeq_All_DE_genes</code></p>
</li>
<li>
<p>Column to use from 2<sup>nd</sup> file</p>
<p>&rarr; <code>1</code></p>
</li>
<li>
<p>Output lines appearing in</p>
<p>&rarr; <code>Both 1st and 2nd files</code></p>
</li>
<li>
<p>First line is a header line</p>
<p>&rarr; <code>Yes</code></p>
</li>
<li>
<p>Ignore case</p>
<p>&rarr; <code>No</code></p>
</li>
<li>
<p>Value to put in unpaired (empty) fields</p>
<p>&rarr; <code>NA</code></p>
</li>
<li>
<p>Click the <code>Run Tool</code> button</p>
</li>
</ul>
</div>
<p><img alt="⚠" class="twemoji" src="https://cdn.jsdelivr.net/gh/jdecked/twemoji@15.0.3/assets/svg/26a0.svg" title=":warning:" /> Rename the output dataset <code>rLog-Normalized counts of DE genes</code></p>
<h3 id="plot-a-heatmap-of-the-rlog-normalized-counts-of-de-genes-in-all-three-conditions">Plot a heatmap of the rLog-Normalized counts of DE genes in all three conditions<a class="headerlink" href="#plot-a-heatmap-of-the-rlog-normalized-counts-of-de-genes-in-all-three-conditions" title="Permanent link">&para;</a></h3>
<p>We do this using the <img align="absbottom" alt="" src="../images/tool_small.png" width="25" /><code>Plot
heatmap with high number of rows</code> tool</p>
<div class="admonition info">
<p class="admonition-title"><img align="absbottom" alt="" src="../images/tool_small.png" width="25" /> <code>Plot heatmap with high number of rows</code> settings</p>
<ul>
<li>
<p>Input should have column headers - these will be the columns that are plotted</p>
<p>&rarr; <code>rLog-Normalized counts of DE genes</code></p>
</li>
<li>
<p>Data transformation</p>
<p>&rarr; <code>Plot the data as it is</code></p>
</li>
<li>
<p>Enable data clustering</p>
<p>&rarr; <code>Yes</code></p>
</li>
<li>
<p>Clustering columns and rows</p>
<p>&rarr; <code>Cluster rows and not columns</code></p>
</li>
<li>
<p>Distance method</p>
<p>&rarr; <code>Euclidean</code></p>
</li>
<li>
<p>Clustering method</p>
<p>&rarr; <code>Complete</code></p>
</li>
<li>
<p>Labeling columns and rows</p>
<p>&rarr; <code>Label columns and not rows</code></p>
</li>
<li>
<p>Coloring groups</p>
<p>&rarr; <code>Blue to white to red</code></p>
</li>
<li>
<p>Data scaling</p>
<p>&rarr; <code>Scale my data by row</code></p>
</li>
<li>
<p>tweak plot height</p>
<p>&rarr; <code>35</code></p>
</li>
<li>
<p>tweak row label size</p>
<p>&rarr; <code>1</code></p>
</li>
<li>
<p>tweak line height</p>
<p>&rarr; <code>24</code></p>
</li>
<li>
<p><code>Run Tool</code></p>
</li>
</ul>
</div>












                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var tab,labels=set.querySelector(".tabbed-labels");for(tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
    
      
      <nav class="md-footer__inner md-grid" aria-label="Footer" >
        
          
          <a href="../27_1_limma/" class="md-footer__link md-footer__link--prev" aria-label="Previous: limma">
            <div class="md-footer__button md-icon">
              
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
            </div>
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Previous
              </span>
              <div class="md-ellipsis">
                limma
              </div>
            </div>
          </a>
        
        
          
          <a href="../28_1_manipulation_edgeR_data/" class="md-footer__link md-footer__link--next" aria-label="Next: edgeR">
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Next
              </span>
              <div class="md-ellipsis">
                edgeR
              </div>
            </div>
            <div class="md-footer__button md-icon">
              
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg>
            </div>
          </a>
        
      </nav>
    
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "../..", "features": ["header.autohide", "navigation.instant", "navigation.tracking", "navigation.tabs", "navigation.tabs.sticky", "navigation.top", "navigation.prune", "navigation.footer", "content.tabs.link", "content.code.copy", "content.action.view", "content.action.edit"], "search": "../../assets/javascripts/workers/search.b8dbb3d2.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.1e8ae164.min.js"></script>
      
        <script src="../../javascripts/mathjax.js"></script>
      
        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
      
        <script src="../../spoiler_files/spoiler.js"></script>
      
    
  </body>
</html>