# Prerequisites

This tutorial aims to help identify cell populations from single cell
transcriptomics data.  

It is composed of R code that you can execute in a R console. You can also 
use any R development environment like Rstudio. You can retrieve the associated
Rscript or R Markdown file (An executable document that combine text and code to
produce formatted report) here : .

!!! info " Installation of R and RStudio "
    Here is some documentation to help you install R and Rstudio if needed :

        - https://larmarange.github.io/analyse-R/installation-de-R-et-RStudio.html
        - https://bookdown.org/introrbook/intro2r/installing-r-and-rstudio.html


# Technical prerequisites

The following points are required to be able to perform the practical
parts of this tutorial correctly :

- [x] Basic knowledge in R because there is no focus on R code
    - How to manipulate  R objets (vectors, dataframe, lists)
    - Create basics functions
    - How to use `lapply` and `mapply`
- [x] R version >= 4 installed
    - You can see all mandatory packages and their version at
     [references](../scRNAseq_basics/references.md) with the `sessionInfo`
