
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://artbio.github.io/startbio/bulk_RNAseq-IOC/32_GOseq/">
      
      
        <link rel="prev" href="../31_GO_enrichment_intro/">
      
      
        <link rel="next" href="../33_exercices_week_05_review/">
      
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.5.3, mkdocs-material-9.5.17">
    
    
      
        <title>GOseq - STARTbio</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.bcfcd587.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.06af60db.min.css">
      
      
  
  
    
    
  
  
  <style>:root{--md-admonition-icon--danger:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 12"><path d="M4.855.708c.5-.896 1.79-.896 2.29 0l4.675 8.351a1.312 1.312 0 0 1-1.146 1.954H1.33A1.313 1.313 0 0 1 .183 9.058ZM7 7V3H5v4Zm-1 3a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z"/></svg>');}</style>



    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Noto+Sans+Display:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Noto Sans Display";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../spoiler_files/spoiler.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="black" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#analysis-of-gene-ontology-enrichments-using-the-galaxy-tool-goseq" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="STARTbio" class="md-header__button md-logo" aria-label="STARTbio" data-md-component="logo">
      
  <img src="../../assets/startbio_icon.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            STARTbio
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              GOseq
            
          </span>
        </div>
      </div>
    </div>
    
      
    
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/ARTbio/startbio" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    ARTbio/startbio
  </div>
</a>
      </div>
    
  </nav>
  
    
      
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../.." class="md-tabs__link">
        
  
    
  
  Home

      </a>
    </li>
  

      
        
  
  
    
  
  
    
    
      
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="../00_IOC_RNAseq_program/" class="md-tabs__link">
          
  
  Interactive Online Companionship (IOC)

        </a>
      </li>
    
  

    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../ten2/Regex_01/" class="md-tabs__link">
          
  
  1Omin2

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../AnalyseGenomes_2025/outline/" class="md-tabs__link">
          
  
  Galaxy Trainings

        </a>
      </li>
    
  

    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../bash_snippets/" class="md-tabs__link">
          
  
  Code Snippets

        </a>
      </li>
    
  

      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../../links/" class="md-tabs__link">
        
  
    
  
  Links

      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
    
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="STARTbio" class="md-nav__button md-logo" aria-label="STARTbio" data-md-component="logo">
      
  <img src="../../assets/startbio_icon.png" alt="logo">

    </a>
    STARTbio
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/ARTbio/startbio" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    ARTbio/startbio
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Home
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
      
        
        
      
      
    
    
      
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" checked>
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="">
            
  
  <span class="md-ellipsis">
    Interactive Online Companionship (IOC)
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            Interactive Online Companionship (IOC)
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
    
  
  
  
    
    
    
      
      
    
    
      
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_1" checked>
        
          
          <label class="md-nav__link" for="__nav_2_1" id="__nav_2_1_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    IOC on bulk RNAseq analysis
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_1_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_2_1">
            <span class="md-nav__icon md-icon"></span>
            IOC on bulk RNAseq analysis
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../00_IOC_RNAseq_program/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Provisional Program / Schedule
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../01_IOC_RNAseq_week_00/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Week 0
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../02_exercices_week_00_review/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Week 1
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../13_exercices_week_01_review/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Week 2
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../19_exercices_week_02_review/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Week 3
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../24_exercices_week_03_review/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Week 4
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

              
            
              
                
  
  
    
  
  
  
    
    
    
      
      
    
    
      
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_1_7" checked>
        
          
          <label class="md-nav__link" for="__nav_2_1_7" id="__nav_2_1_7_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Week 5
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_2_1_7_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_2_1_7">
            <span class="md-nav__icon md-icon"></span>
            Week 5
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../30_exercices_week_04_review/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Review on week-4 work
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../31_GO_enrichment_intro/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Gene Ontology Enrichment Analysis
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    
    
    
      
      
    
    
      
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_1_7_3" checked>
        
          
          <label class="md-nav__link" for="__nav_2_1_7_3" id="__nav_2_1_7_3_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Week 5 exercices
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_2_1_7_3_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_2_1_7_3">
            <span class="md-nav__icon md-icon"></span>
            Week 5 exercices
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    GOseq
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    GOseq
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#goseq-analysis-of-the-deseq2-de-tables-in-the-use-case-prjna630433" class="md-nav__link">
    <span class="md-ellipsis">
      goseq analysis of the DESeq2 DE tables in the use-case PRJNA630433
    </span>
  </a>
  
    <nav class="md-nav" aria-label="goseq analysis of the DESeq2 DE tables in the use-case PRJNA630433">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#gather-needed-data-in-a-new-history" class="md-nav__link">
    <span class="md-ellipsis">
      Gather needed data in a new history
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#prepare-the-gene-sets" class="md-nav__link">
    <span class="md-ellipsis">
      Prepare the Gene Set(s)
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Prepare the Gene Set(s)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#clean-up-tables-from-nas" class="md-nav__link">
    <span class="md-ellipsis">
      Clean up tables from NAs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#add-a-boolean-column-to-tag-the-gene-set" class="md-nav__link">
    <span class="md-ellipsis">
      Add a boolean column to tag the gene Set
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#cut-columns-1-and-8-to-get-the-final-gene-sets-and-backgrounds" class="md-nav__link">
    <span class="md-ellipsis">
      Cut columns 1 and 8 to get the final gene sets and backgrounds
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#extract-a-single-gene-length-table-dataset" class="md-nav__link">
    <span class="md-ellipsis">
      Extract a single gene length table dataset
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#perform-goseq-analysis" class="md-nav__link">
    <span class="md-ellipsis">
      Perform goseq analysis
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../33_exercices_week_05_review/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Week 6
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../40_exercices_week_06_review/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Week 7
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../50_exercices_week_07_review/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Week 8
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../links/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    External Materials
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    
  
  
    <a href="../../R-IOC/00_IOC_R_program/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    IOC on R
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

  

      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    
  
  
    <a href="../../scRNAseq_basics/program_sc/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    IOC on Single-cell RNAseq analysis using Seurat in R
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

  

      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../../deepseq/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Introduction to Deep-seq Analysis
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../../ten2/Regex_01/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    1Omin2
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    
  
  
    <a href="../../AnalyseGenomes_2025/outline/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Galaxy Trainings
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

  

      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../../bash_snippets/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Code Snippets
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../links/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Links
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#goseq-analysis-of-the-deseq2-de-tables-in-the-use-case-prjna630433" class="md-nav__link">
    <span class="md-ellipsis">
      goseq analysis of the DESeq2 DE tables in the use-case PRJNA630433
    </span>
  </a>
  
    <nav class="md-nav" aria-label="goseq analysis of the DESeq2 DE tables in the use-case PRJNA630433">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#gather-needed-data-in-a-new-history" class="md-nav__link">
    <span class="md-ellipsis">
      Gather needed data in a new history
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#prepare-the-gene-sets" class="md-nav__link">
    <span class="md-ellipsis">
      Prepare the Gene Set(s)
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Prepare the Gene Set(s)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#clean-up-tables-from-nas" class="md-nav__link">
    <span class="md-ellipsis">
      Clean up tables from NAs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#add-a-boolean-column-to-tag-the-gene-set" class="md-nav__link">
    <span class="md-ellipsis">
      Add a boolean column to tag the gene Set
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#cut-columns-1-and-8-to-get-the-final-gene-sets-and-backgrounds" class="md-nav__link">
    <span class="md-ellipsis">
      Cut columns 1 and 8 to get the final gene sets and backgrounds
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#extract-a-single-gene-length-table-dataset" class="md-nav__link">
    <span class="md-ellipsis">
      Extract a single gene length table dataset
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#perform-goseq-analysis" class="md-nav__link">
    <span class="md-ellipsis">
      Perform goseq analysis
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
    <a href="https://github.com/ARTbio/startbio/edit/master/docs/bulk_RNAseq-IOC/32_GOseq.md" title="Edit this page" class="md-content__button md-icon">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M10 20H6V4h7v5h5v3.1l2-2V8l-6-6H6c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h4v-2m10.2-7c.1 0 .3.1.4.2l1.3 1.3c.2.2.2.6 0 .8l-1 1-2.1-2.1 1-1c.1-.1.2-.2.4-.2m0 3.9L14.1 23H12v-2.1l6.1-6.1 2.1 2.1Z"/></svg>
    </a>
  
  
    
      
    
    <a href="https://github.com/ARTbio/startbio/raw/master/docs/bulk_RNAseq-IOC/32_GOseq.md" title="View source of this page" class="md-content__button md-icon">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 18c.56 0 1 .44 1 1s-.44 1-1 1-1-.44-1-1 .44-1 1-1m0-3c-2.73 0-5.06 1.66-6 4 .94 2.34 3.27 4 6 4s5.06-1.66 6-4c-.94-2.34-3.27-4-6-4m0 6.5a2.5 2.5 0 0 1-2.5-2.5 2.5 2.5 0 0 1 2.5-2.5 2.5 2.5 0 0 1 2.5 2.5 2.5 2.5 0 0 1-2.5 2.5M9.27 20H6V4h7v5h5v4.07c.7.08 1.36.25 2 .49V8l-6-6H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h4.5a8.15 8.15 0 0 1-1.23-2Z"/></svg>
    </a>
  


<h1 id="analysis-of-gene-ontology-enrichments-using-the-galaxy-tool-goseq">Analysis of Gene Ontology enrichments using the Galaxy Tool <code>goseq</code><a class="headerlink" href="#analysis-of-gene-ontology-enrichments-using-the-galaxy-tool-goseq" title="Permanent link">&para;</a></h1>
<p>As mentioned previously NGS read count data are inherently biased by the length of
transcripts.</p>
<p>How does this bias translate in DE tables ?</p>
<p><mark>Not</mark> in biased Fold Changes ! Fold changes are count ratios for a given gene, so bias
is compensated for.</p>
<p>In contrast, genes with a high number of counts (whether due to a high level of expression
or a large transcript) tend to be <mark>over-<strong>detected</strong></mark> since the p-value returned by the
statistical test depends on the count base mean. This becomes a major issue in GO-based
Gene Set Analyses since gene sets are selected on the basis of their p-values/p-adjusted-values.</p>
<p>The <code>goseq</code> tool provides methods for performing GO analysis of RNA-seq data, taking
length bias into account.</p>
<h2 id="goseq-analysis-of-the-deseq2-de-tables-in-the-use-case-prjna630433"><code>goseq</code> analysis of the DESeq2 DE tables in the use-case PRJNA630433<a class="headerlink" href="#goseq-analysis-of-the-deseq2-de-tables-in-the-use-case-prjna630433" title="Permanent link">&para;</a></h2>
<h3 id="gather-needed-data-in-a-new-history">Gather needed data in a new history<a class="headerlink" href="#gather-needed-data-in-a-new-history" title="Permanent link">&para;</a></h3>
<ul class="task-list">
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> Although we are going to focus on the DESeq2 tables, the approach would be the same
with other DE callers: feel free to test it latter on.</li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> to perform the analysis, we will need a gene length list. This list was generated
<a href="../21_FeatureCounts/">previously by the featureCounts tool</a> and is therefore available in
the history <code>PRJNA630433 FeatureCounts Counting on HISAT2 bam alignments</code>.</li>
</ul>
<p>&rarr; Copy one of the collection among the <code>Dc</code>, <code>Mo</code> or <code>Oc FeatureCounts Feature Length</code>
  collections in this history in a new history which you will name <code>PRJNA630433 GOseq
  analysis</code>. Note that Feature lenght datasets are all identical, we will just need to
  extract one for the goseq analysis.</p>
<ul class="task-list">
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> Finally, copy the collection of DEseq2 tables from the history <code>PRJNA630433 DESeq2
  analysis</code> in the new history <code>PRJNA630433 FeatureCounts Counting on HISAT2 bam
  alignments</code>. This collection should be named <code>DESeq2 Results Tables</code>. </li>
</ul>
<h3 id="prepare-the-gene-sets">Prepare the Gene Set(s)<a class="headerlink" href="#prepare-the-gene-sets" title="Permanent link">&para;</a></h3>
<p>Since we are going to perform a GO-based Gene Set Analysis we first need to define the
Gene Sets ! We will keept the advantage of collections and treat in parallel the three
comparisons (Mo vs Dc, Oc vs Dc and Oc vs Mo).</p>
<h4 id="clean-up-tables-from-nas">Clean up tables from <code>NA</code>s<a class="headerlink" href="#clean-up-tables-from-nas" title="Permanent link">&para;</a></h4>
<p>We first clean up the Tables by removing all lines that contain <code>NA</code> values using the Tool
<code>Select lines that match an expression</code></p>
<p>This is an important step because it allows to reduce the "gene space". As mentioned in
introduction, choosing a relevant background list representing the genes <mark>not
differentially expressed</mark> is crucial for accurate enrichment analysis. By removing the
lines containing NAs, we shrink our datasets and retains only genes that are accessible to
tests, ie, that are at least significantly <mark>expressed</mark> in the considered tissue/experience.</p>
<div class="admonition info">
<p class="admonition-title"><img align="absbottom" alt="" src="../images/tool_small.png" width="25" /> <code>Select lines that match an expression</code> settings</p>
<ul>
<li>
<p>Select lines from</p>
<p>&rarr; Click the collection icon and select <code>DESeq2 Results Tables</code></p>
</li>
<li>
<p>that</p>
<p>&rarr; <code>Not matching</code> (be careful, <mark>not</mark> matching)</p>
</li>
<li>
<p>the pattern</p>
<p>&rarr; <code>\tNA$</code> (we seach for lines that contain any tabulation mark, followed by <code>NA</code> and
this <code>NA</code> is a word, ie, it is followed by a space or another tabulation mark)</p>
</li>
<li>
<p>Keep header line</p>
<p>&rarr; <code>Yes</code></p>
</li>
<li>
<p><code>Run Tool</code></p>
</li>
</ul>
</div>
<h4 id="add-a-boolean-column-to-tag-the-gene-set">Add a boolean column to tag the gene Set<a class="headerlink" href="#add-a-boolean-column-to-tag-the-gene-set" title="Permanent link">&para;</a></h4>
<p>Now we are going to select our gene sets from DE table collection using a treatment which
evaluates a each line of the table and returns a boolean value in a new column. Genes with
a <code>True</code> value will been considered as part of the gene set, whereas genes with a <code>False</code>
value will be considered as part of the background list for the GSA.</p>
<p>Let's do this, using a stringent evaluation, ie we will tag genes for which both <code>p-adjust
&lt; 0.001</code> and <code>|log2FC| &gt; 2</code>. In your own analyses you may have to use less stringent
filtering. It depends mostly of your datasets and whether you get a sufficiently large enough
gene set after filtering. Considere that the gene set size should be between 1 and 10 % of
your background genes.</p>
<div class="admonition info">
<p class="admonition-title"><img align="absbottom" alt="" src="../images/tool_small.png" width="25" /> <code>Compute on rows</code> settings</p>
<ul>
<li>
<p>Input file</p>
<p>&rarr; Click the collection icon and select <code>Select on collection 9</code></p>
</li>
<li>
<p>Input has a header line with column names?</p>
<p>&rarr; <code>Yes</code></p>
</li>
<li>
<p>Add expression</p>
<p>&rarr; <code>c7 &lt; 0.001 and abs(c3) &gt; 2</code></p>
</li>
<li>
<p>Mode of the operation</p>
<p>&rarr; <code>Append</code></p>
</li>
<li>
<p>The new column name</p>
<p>&rarr; <code>boolean</code></p>
</li>
<li>
<p><code>Run Tool</code></p>
</li>
</ul>
</div>
<h4 id="cut-columns-1-and-8-to-get-the-final-gene-sets-and-backgrounds">Cut columns 1 and 8 to get the final gene sets and backgrounds<a class="headerlink" href="#cut-columns-1-and-8-to-get-the-final-gene-sets-and-backgrounds" title="Permanent link">&para;</a></h4>
<div class="admonition info">
<p class="admonition-title"><img align="absbottom" alt="" src="../images/tool_small.png" width="25" /> <code>Advanced Cut columns from a table</code> settings</p>
<ul>
<li>
<p>File to cut</p>
<p>&rarr; Click the collection icon and select <code>Compute on collection 10</code></p>
</li>
<li>
<p>Operation</p>
<p>&rarr; <code>Keep</code></p>
</li>
<li>
<p>Delimited by</p>
<p>&rarr; <code>Tab</code></p>
</li>
<li>
<p>Cut by</p>
<p>&rarr; <code>fields</code></p>
</li>
<li>
<p>List of Fields</p>
<p>&rarr; <code>column 1</code> and <code>column 3</code></p>
</li>
<li>
<p><code>Run Tool</code></p>
</li>
</ul>
</div>
<p><img alt="⚠" class="twemoji" src="https://cdn.jsdelivr.net/gh/jdecked/twemoji@15.0.3/assets/svg/26a0.svg" title=":warning:" /> As this is the last step of the construction of the gene set lists, you should
rename for clarity the returned collection as <code>Gene Lists</code>.</p>
<h3 id="extract-a-single-gene-length-table-dataset">Extract a single gene length table dataset<a class="headerlink" href="#extract-a-single-gene-length-table-dataset" title="Permanent link">&para;</a></h3>
<p>A single gene length table can be extract from the collection which we initially copied from
the history <code>PRJNA630433 FeatureCounts Counting on HISAT2 bam alignments</code></p>
<div class="admonition info">
<p class="admonition-title"><img align="absbottom" alt="" src="../images/tool_small.png" width="25" /> <code>Extract dataset</code> settings</p>
<ul>
<li>
<p>Input List</p>
<p>&rarr; Click the collection icon and select <code>Mo featureCounts: Feature lengths</code></p>
</li>
<li>
<p>How should a dataset be selected?</p>
<p>&rarr; <code>The first dataset</code></p>
<p><img alt="⚠" class="twemoji" src="https://cdn.jsdelivr.net/gh/jdecked/twemoji@15.0.3/assets/svg/26a0.svg" title=":warning:" /> it would work equally well with <code>Select by element identifier</code> or
<code>Select by index</code></p>
</li>
<li>
<p><code>Run Tool</code></p>
</li>
</ul>
</div>
<p><img alt="⚠" class="twemoji" src="https://cdn.jsdelivr.net/gh/jdecked/twemoji@15.0.3/assets/svg/26a0.svg" title=":warning:" /> For clarity, also rename this single dataset as <code>Gene lengths</code></p>
<h3 id="perform-goseq-analysis">Perform goseq analysis<a class="headerlink" href="#perform-goseq-analysis" title="Permanent link">&para;</a></h3>
<p>Our two inputs are now ready for goseq:
- The list of tagged genes. <code>True</code> for the gene set, <code>False</code> for the background genes
- The length of genes (aggregated size of exons) to correct for the length bias of detection.</p>
<p>goseq proposes 3 types of GSA methods</p>
<ul class="task-list">
<li class="task-list-item">
<p><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> The <strong>Wallenius method</strong> approximates the true distribution of numbers of members
  of a category amongst DE genes by the Wallenius non-central hypergeometric distribution.
  This distribution assumes that within a category all genes have the same probability of
  being chosen. Therefore, this approximation works best when the range in probabilities
  obtained by the probability weighting function (the regression over gene lenght) is small.
  This is the method specifically developed in the goseq package as well as the one we are
  going to choose in this IOC.</p>
</li>
<li class="task-list-item">
<p><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> The <strong>Sampling method</strong> uses random sampling to approximate the true distribution
  and uses it to calculate the p-values for over (and under) representation of categories.
  Although this is the most accurate method given a high enough value of sampling number,
  its use quickly becomes computationally prohibitive. It may sometimes be desirable to
  use random sampling to generate the null distribution for category membership. For
  example, to check consistency against results from the Wallenius approximation. This is
  easily accomplished by using the method option to additionally specify sampling and the
  number of samples to generate.</p>
</li>
<li class="task-list-item">
<p><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> The <strong>Hypergeometric Method</strong> assumes <mark>there is no bias</mark> in power to detect
  differential expression at all and calculates the p-values using a standard hypergeometric
  distribution (no length bias correction is performed). Useful if you wish to test the
  effect of length bias on your results. <img alt="⚠" class="twemoji" src="https://cdn.jsdelivr.net/gh/jdecked/twemoji@15.0.3/assets/svg/26a0.svg" title=":warning:" /> Hypergeometric method should NEVER be
  used for producing results for biological interpretation of RNA-seq data. Indeed, if
  length bias is truly not present in your data, goseq will produce a nearly flat PWF plot,
  no length bias correction will be applied to your data, and all methods will produce the
  same results.</p>
</li>
</ul>
<div class="admonition info">
<p class="admonition-title"><img align="absbottom" alt="" src="../images/tool_small.png" width="25" /> <code>goseq tests</code> settings</p>
<ul>
<li>
<p>Differentially expressed genes file</p>
<p>&rarr; Click the collection icon and select <code>Gene Lists</code> (the renamed collection)</p>
</li>
<li>
<p>Gene lengths file</p>
<p>&rarr; <code>Gene lengths</code> (the renamed single dataset)</p>
</li>
<li>
<p>Gene categories</p>
<p>Because we are working here with Mus musculus, we can rely on the tool to fetch
directly the GO categories from a remote database. <img alt="⚠" class="twemoji" src="https://cdn.jsdelivr.net/gh/jdecked/twemoji@15.0.3/assets/svg/26a0.svg" title=":warning:" /> For <em>Nesseria gon</em> As
well as <em>Apis mel</em> you will have to construct and use your own Gene categorie file.
And... Yes, the section title <code>Gene categories</code> is not sufficiently explicit and
should rather be <code>GO categories</code></p>
<p>&rarr; <code>Get categories</code></p>
</li>
<li>
<p>Select a genome to use</p>
<p>&rarr; <code>Mouse (mm10)</code></p>
</li>
<li>
<p>Select Gene ID format</p>
<p>&rarr; <code>Ensembl Gene ID</code></p>
</li>
<li>
<p>Select one or more categories</p>
<p>&rarr; Check only <code>GO: Biological Process</code> (for simplicity in this first run)</p>
</li>
<li>
<p>Method Options</p>
<p>&rarr; Use Wallenius method <code>Yes</code></p>
<p>&rarr; Use Hypergeometric method <code>No</code></p>
<p>&rarr; Sampling number <code>5000</code> <img alt="⚠" class="twemoji" src="https://cdn.jsdelivr.net/gh/jdecked/twemoji@15.0.3/assets/svg/26a0.svg" title=":warning:" /> We also run this method to compare it with
the Wallenius method.</p>
</li>
<li>
<p>Advanced Options </p>
<p>&rarr; Select a method for multiple hypothesis testing correction <code>Benjamini-Hochberg [FDR]</code></p>
<p>&rarr; Count genes without any category? <code>Yes</code></p>
</li>
<li>
<p>Output Options</p>
<p>&rarr; Output Top GO terms plot? <code>Yes</code></p>
<p>&rarr; Produce diagnostic plots? <code>Yes</code></p>
<p>&rarr; Extract the DE genes for the categories (GO/KEGG terms)? <code>Yes</code></p>
</li>
<li>
<p><code>Run Tool</code></p>
</li>
</ul>
</div>
<p><code>goseq</code> generates a big table with the following columns for each GO term:</p>
<table>
<thead>
<tr>
<th>Column</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>category</td>
<td>GO category</td>
</tr>
<tr>
<td>over_rep_pval</td>
<td>p-value for over representation of the term in the differentially expressed genes</td>
</tr>
<tr>
<td>under_rep_pval</td>
<td>p-value for under representation of the term in the differentially expressed genes</td>
</tr>
<tr>
<td>numDEInCat</td>
<td>number of differentially expressed genes in this category</td>
</tr>
<tr>
<td>numInCat</td>
<td>number of genes in this category</td>
</tr>
<tr>
<td>term</td>
<td>detail of the term</td>
</tr>
<tr>
<td>ontology</td>
<td>MF (Molecular Function - molecular activities of gene products), CC (Cellular Component - where gene products are active), BP (Biological Process - pathways and larger processes made up of the activities of multiple gene products)</td>
</tr>
<tr>
<td>p.adjust.over_represented</td>
<td>p-value for over representation of the term in the differentially expressed genes, adjusted for multiple testing with the Benjamini-Hochberg procedure</td>
</tr>
<tr>
<td>p.adjust.under_represented</td>
<td>p-value for over representation of the term in the differentially expressed genes, adjusted for multiple testing with the Benjamini-Hochberg procedure</td>
</tr>
</tbody>
</table>
<p>To identify categories significantly enriched/unenriched below some p-value cutoff, it is necessary to use the adjusted p-value.</p>
<details class="question">
<summary>How many GO terms are <strong>over</strong>-represented at adjusted P value &lt; 0.05?</summary>
</details>
<details class="question">
<summary>How many GO terms are <strong>under</strong>-represented at adjusted P value &lt; 0.05?</summary>
</details>












                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var tab,labels=set.querySelector(".tabbed-labels");for(tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
    
      
      <nav class="md-footer__inner md-grid" aria-label="Footer" >
        
          
          <a href="../31_GO_enrichment_intro/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Gene Ontology Enrichment Analysis">
            <div class="md-footer__button md-icon">
              
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
            </div>
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Previous
              </span>
              <div class="md-ellipsis">
                Gene Ontology Enrichment Analysis
              </div>
            </div>
          </a>
        
        
          
          <a href="../33_exercices_week_05_review/" class="md-footer__link md-footer__link--next" aria-label="Next: Review on week-5 work">
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Next
              </span>
              <div class="md-ellipsis">
                Review on week-5 work
              </div>
            </div>
            <div class="md-footer__button md-icon">
              
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg>
            </div>
          </a>
        
      </nav>
    
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "../..", "features": ["header.autohide", "navigation.instant", "navigation.tracking", "navigation.tabs", "navigation.tabs.sticky", "navigation.top", "navigation.prune", "navigation.footer", "content.tabs.link", "content.code.copy", "content.action.view", "content.action.edit"], "search": "../../assets/javascripts/workers/search.b8dbb3d2.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.1e8ae164.min.js"></script>
      
        <script src="../../javascripts/mathjax.js"></script>
      
        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
      
        <script src="../../spoiler_files/spoiler.js"></script>
      
    
  </body>
</html>