In this training, you are going to

1. Deploy your own Galaxy server in a Google Virtual machine
      
      :warning: ==This will be done outside of class hours== as it takes
      computer time, during which you cannot do analyses.
      
- Review the operations to stop, start, restart your Galaxy server in your Google
  virtual machine

- Prepare your Galaxy server for the next day analyses
      
      Upload references data and sequencing datasets, index genome references for fast read
      alignments, etc.
      
- Compare the procedures for performing a simple file formatting task, either using command
  lines in the linux environment or using Galaxy
     
     This will provide also the opportunity to take the control of your Galaxy server deployed
     in `1.`


By the end of the training session, you should have prepared your Galaxy analysis environment
and get it ready to perform subsequent analyses of small RNAseq and RNAseq datasets as
described [here](https://slecrom.github.io/AG2023/)
