In this training, we are going to

1. Deploy your own Galaxy server in a Google Virtual machine
      
      We start by this because it takes a bit of computer time
      
- Compare the procedures for performing a simple file formatting task, either using command
  lines in the linux environment or using Galaxy
     
     This will provide also the opportunity to take the control of your Galaxy server deployed
     in `1.`

- Prepare your Galaxy server for the next day analyses
      
      Upload references data and sequencing datasets, index genome references for fast read
      alignments, etc.
      
- Review together the operations to stop, start, restart your Galaxy server in your Google
  virtual machine

By the end of the training session, you should have prepared your Galaxy analysis environment
and get it ready to perform subsequent analyses of small RNAseq and RNAseq datasets as
described [here](https://slecrom.github.io/AG2023/)
