
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://artbio.github.io/startbio/bulk_RNAseq-IOC/06_sequence_quality_read_filtering/">
      
      
        <link rel="prev" href="../05_sequencing_strategies/">
      
      
        <link rel="next" href="../07_transcript_quant/">
      
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.5.3, mkdocs-material-9.5.17">
    
    
      
        <title>Sequence Quality and Read Filtering - STARTbio</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.bcfcd587.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.06af60db.min.css">
      
      
  
  
    
    
  
  
  <style>:root{--md-admonition-icon--danger:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 12"><path d="M4.855.708c.5-.896 1.79-.896 2.29 0l4.675 8.351a1.312 1.312 0 0 1-1.146 1.954H1.33A1.313 1.313 0 0 1 .183 9.058ZM7 7V3H5v4Zm-1 3a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z"/></svg>');}</style>



    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Noto+Sans+Display:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Noto Sans Display";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../spoiler_files/spoiler.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="black" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#fastq-files-and-fastq-quality" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="STARTbio" class="md-header__button md-logo" aria-label="STARTbio" data-md-component="logo">
      
  <img src="../../assets/startbio_icon.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            STARTbio
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Sequence Quality and Read Filtering
            
          </span>
        </div>
      </div>
    </div>
    
      
    
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/ARTbio/startbio" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    ARTbio/startbio
  </div>
</a>
      </div>
    
  </nav>
  
    
      
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../.." class="md-tabs__link">
        
  
    
  
  Home

      </a>
    </li>
  

      
        
  
  
    
  
  
    
    
      
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="../00_IOC_RNAseq_program/" class="md-tabs__link">
          
  
  Interactive Online Companionship (IOC)

        </a>
      </li>
    
  

    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../ten2/Regex_01/" class="md-tabs__link">
          
  
  1Omin2

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../AnalyseGenomes_2025/outline/" class="md-tabs__link">
          
  
  Galaxy Trainings

        </a>
      </li>
    
  

    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../bash_snippets/" class="md-tabs__link">
          
  
  Code Snippets

        </a>
      </li>
    
  

      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../../links/" class="md-tabs__link">
        
  
    
  
  Links

      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
    
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="STARTbio" class="md-nav__button md-logo" aria-label="STARTbio" data-md-component="logo">
      
  <img src="../../assets/startbio_icon.png" alt="logo">

    </a>
    STARTbio
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/ARTbio/startbio" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    ARTbio/startbio
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Home
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
      
        
        
      
      
    
    
      
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" checked>
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="">
            
  
  <span class="md-ellipsis">
    Interactive Online Companionship (IOC)
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            Interactive Online Companionship (IOC)
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
    
  
  
  
    
    
    
      
      
    
    
      
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_1" checked>
        
          
          <label class="md-nav__link" for="__nav_2_1" id="__nav_2_1_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    IOC on bulk RNAseq analysis
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_1_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_2_1">
            <span class="md-nav__icon md-icon"></span>
            IOC on bulk RNAseq analysis
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../00_IOC_RNAseq_program/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Provisional Program / Schedule
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../01_IOC_RNAseq_week_00/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Week 0
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

              
            
              
                
  
  
    
  
  
  
    
    
    
      
      
    
    
      
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_1_3" checked>
        
          
          <label class="md-nav__link" for="__nav_2_1_3" id="__nav_2_1_3_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Week 1
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="3" aria-labelledby="__nav_2_1_3_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_2_1_3">
            <span class="md-nav__icon md-icon"></span>
            Week 1
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../02_exercices_week_00_review/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Review on week-0 work
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    
    
    
      
      
    
    
      
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_1_3_2" checked>
        
          
          <label class="md-nav__link" for="__nav_2_1_3_2" id="__nav_2_1_3_2_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Analysis Overwiew
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="4" aria-labelledby="__nav_2_1_3_2_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_2_1_3_2">
            <span class="md-nav__icon md-icon"></span>
            Analysis Overwiew
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../03_readcounts/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    The Key idea
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../04_cDNAs/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    RNAseq librairies
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    Sequence Quality and Read Filtering
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    Sequence Quality and Read Filtering
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#fastq-files-and-fastq-quality" class="md-nav__link">
    <span class="md-ellipsis">
      Fastq files and Fastq quality
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Fastq files and Fastq quality">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#fastq-format" class="md-nav__link">
    <span class="md-ellipsis">
      FASTQ format
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#quality-score-encoding" class="md-nav__link">
    <span class="md-ellipsis">
      Quality Score Encoding
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#assessing-quality-with-fastqc" class="md-nav__link">
    <span class="md-ellipsis">
      Assessing quality with FastQC
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#quick-discussion-of-quality-metrics-assessed-by-fastqc" class="md-nav__link">
    <span class="md-ellipsis">
      Quick discussion of quality metrics assessed by FastQC
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#read-filtering" class="md-nav__link">
    <span class="md-ellipsis">
      Read filtering
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../07_transcript_quant/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Transcript Quantification
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../08_RNAseq_DE/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Statistical Analysis of Differential expression
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../09_outline_conclusion/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Understanding of all your experimental procedures
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    
  
  
    <a href="../10_PRJNA630433_presentation/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Week 1 exercises
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

  

      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../13_exercices_week_01_review/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Week 2
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../19_exercices_week_02_review/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Week 3
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../24_exercices_week_03_review/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Week 4
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../30_exercices_week_04_review/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Week 5
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../33_exercices_week_05_review/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Week 6
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../40_exercices_week_06_review/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Week 7
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../50_exercices_week_07_review/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Week 8
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../links/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    External Materials
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    
  
  
    <a href="../../R-IOC/00_IOC_R_program/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    IOC on R
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

  

      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    
  
  
    <a href="../../scRNAseq_basics/program_sc/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    IOC on Single-cell RNAseq analysis using Seurat in R
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

  

      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../../deepseq/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Introduction to Deep-seq Analysis
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../../ten2/Regex_01/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    1Omin2
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    
  
  
    <a href="../../AnalyseGenomes_2025/outline/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Galaxy Trainings
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

  

      
    </li>
  

    
      
      
  
  
  
  
    
    
    
      
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../../bash_snippets/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Code Snippets
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../links/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Links
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#fastq-files-and-fastq-quality" class="md-nav__link">
    <span class="md-ellipsis">
      Fastq files and Fastq quality
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Fastq files and Fastq quality">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#fastq-format" class="md-nav__link">
    <span class="md-ellipsis">
      FASTQ format
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#quality-score-encoding" class="md-nav__link">
    <span class="md-ellipsis">
      Quality Score Encoding
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#assessing-quality-with-fastqc" class="md-nav__link">
    <span class="md-ellipsis">
      Assessing quality with FastQC
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#quick-discussion-of-quality-metrics-assessed-by-fastqc" class="md-nav__link">
    <span class="md-ellipsis">
      Quick discussion of quality metrics assessed by FastQC
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#read-filtering" class="md-nav__link">
    <span class="md-ellipsis">
      Read filtering
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
    <a href="https://github.com/ARTbio/startbio/edit/master/docs/bulk_RNAseq-IOC/06_sequence_quality_read_filtering.md" title="Edit this page" class="md-content__button md-icon">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M10 20H6V4h7v5h5v3.1l2-2V8l-6-6H6c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h4v-2m10.2-7c.1 0 .3.1.4.2l1.3 1.3c.2.2.2.6 0 .8l-1 1-2.1-2.1 1-1c.1-.1.2-.2.4-.2m0 3.9L14.1 23H12v-2.1l6.1-6.1 2.1 2.1Z"/></svg>
    </a>
  
  
    
      
    
    <a href="https://github.com/ARTbio/startbio/raw/master/docs/bulk_RNAseq-IOC/06_sequence_quality_read_filtering.md" title="View source of this page" class="md-content__button md-icon">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 18c.56 0 1 .44 1 1s-.44 1-1 1-1-.44-1-1 .44-1 1-1m0-3c-2.73 0-5.06 1.66-6 4 .94 2.34 3.27 4 6 4s5.06-1.66 6-4c-.94-2.34-3.27-4-6-4m0 6.5a2.5 2.5 0 0 1-2.5-2.5 2.5 2.5 0 0 1 2.5-2.5 2.5 2.5 0 0 1 2.5 2.5 2.5 2.5 0 0 1-2.5 2.5M9.27 20H6V4h7v5h5v4.07c.7.08 1.36.25 2 .49V8l-6-6H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h4.5a8.15 8.15 0 0 1-1.23-2Z"/></svg>
    </a>
  


  <h1>Sequence Quality and Read Filtering</h1>

<h2 id="fastq-files-and-fastq-quality">Fastq files and Fastq quality<a class="headerlink" href="#fastq-files-and-fastq-quality" title="Permanent link">&para;</a></h2>
<p>Next generation sequencing report sequences in the FASTQ format.</p>
<p>In this FASQTQ format, both the sequence letter and quality score are each encoded with a
single ASCII character for brevity and for alignment between a reported nucleotide and
the quality of its calling.</p>
<h3 id="fastq-format">FASTQ format<a class="headerlink" href="#fastq-format" title="Permanent link">&para;</a></h3>
<p>A FASTQ file has four line-separated fields per sequence:</p>
<ul>
<li>Line 1 begins with a '@' character and is followed by a sequence identifier and an optional description (like a FASTA title line).</li>
<li>Line 2 is the raw sequence letters.</li>
<li>Line 3 begins with a '+' character and is optionally followed by the same sequence identifier (and any description) again.</li>
<li>Line 4 encodes the quality values for the sequence in Field 2, and must contain the same number of symbols as letters in the sequence.</li>
</ul>
<div class="admonition info">
<p class="admonition-title">A FASTQ file containing a single sequence might look like this:</p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-0-1"><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a>@A00680:51:HF2TKDRXX:2:2133:6596:12289/1
</span><span id="__span-0-2"><a id="__codelineno-0-2" name="__codelineno-0-2" href="#__codelineno-0-2"></a>AGAGTAAGTCTTTGTATTTTATGCTACTGTACCTCTGGGATTAATTGCTC
</span><span id="__span-0-3"><a id="__codelineno-0-3" name="__codelineno-0-3" href="#__codelineno-0-3"></a>+
</span><span id="__span-0-4"><a id="__codelineno-0-4" name="__codelineno-0-4" href="#__codelineno-0-4"></a>BFCHBBEGAHDDDHABCDDDBCHFDCFDHABGEDGDHGGCCDBECEHFDG
</span><span id="__span-0-5"><a id="__codelineno-0-5" name="__codelineno-0-5" href="#__codelineno-0-5"></a>              123
</span></code></pre></div>
<table>
<thead>
<tr>
<th>Q letter</th>
<th>Q score</th>
<th>Prob(Err)</th>
</tr>
</thead>
<tbody>
<tr>
<td>A (Pos 1)</td>
<td>32</td>
<td>0.00063</td>
</tr>
<tr>
<td>B (Pos 2)</td>
<td>33</td>
<td>0.0005</td>
</tr>
<tr>
<td>C (Pos 3)</td>
<td>34</td>
<td>0.00039</td>
</tr>
</tbody>
</table>
</div>
<p>The quality score (or Q-score) expresses an error probability. In particular, it serves as
a convenient and compact way to communicate very small error probabilities.</p>
<p>Given an assertion, A, the quality score, Q(A), expresses the probability that A is not true,
P(~A), according to the relationship:</p>
<p><code>Q(A) =-10 log10(P(~A))</code></p>
<p>The relationship between the quality score and error probability is demonstrated with the
following table:</p>
<table>
<thead>
<tr>
<th>Quality score, Q(A)</th>
<th>Error probability, P(~A)</th>
</tr>
</thead>
<tbody>
<tr>
<td>10</td>
<td>0.1</td>
</tr>
<tr>
<td>20</td>
<td>0.01</td>
</tr>
<tr>
<td>30</td>
<td>0.001</td>
</tr>
</tbody>
</table>
<h3 id="quality-score-encoding">Quality Score Encoding<a class="headerlink" href="#quality-score-encoding" title="Permanent link">&para;</a></h3>
<p>In FASTQ files, quality scores are encoded into a compact form, which uses only 1 byte per
quality value. In this encoding, the quality score is represented as the character with an
ASCII code equal to its value + 33 (offset of +33). The following table demonstrates the
relationship between the encoding character, its ASCII code, and the quality score represented.</p>
<details class="info">
<summary>relationship between the encoding character, its ASCII code, and the quality score represented</summary>
<table>
<thead>
<tr>
<th>Symbol</th>
<th>ASCII Code</th>
<th>Q-Score</th>
</tr>
</thead>
<tbody>
<tr>
<td>!</td>
<td>33</td>
<td>0</td>
</tr>
<tr>
<td>"</td>
<td>34</td>
<td>1</td>
</tr>
<tr>
<td>#</td>
<td>35</td>
<td>2</td>
</tr>
<tr>
<td>$</td>
<td>36</td>
<td>3</td>
</tr>
<tr>
<td>%</td>
<td>37</td>
<td>4</td>
</tr>
<tr>
<td>&amp;</td>
<td>38</td>
<td>5</td>
</tr>
<tr>
<td>'</td>
<td>39</td>
<td>6</td>
</tr>
<tr>
<td>(</td>
<td>40</td>
<td>7</td>
</tr>
<tr>
<td>)</td>
<td>41</td>
<td>8</td>
</tr>
<tr>
<td>*</td>
<td>42</td>
<td>9</td>
</tr>
<tr>
<td>+</td>
<td>43</td>
<td>10</td>
</tr>
<tr>
<td>,</td>
<td>44</td>
<td>11</td>
</tr>
<tr>
<td>-</td>
<td>45</td>
<td>12</td>
</tr>
<tr>
<td>.</td>
<td>46</td>
<td>13</td>
</tr>
<tr>
<td>/</td>
<td>47</td>
<td>14</td>
</tr>
<tr>
<td>0</td>
<td>48</td>
<td>15</td>
</tr>
<tr>
<td>1</td>
<td>49</td>
<td>16</td>
</tr>
<tr>
<td>2</td>
<td>50</td>
<td>17</td>
</tr>
<tr>
<td>3</td>
<td>51</td>
<td>18</td>
</tr>
<tr>
<td>4</td>
<td>52</td>
<td>19</td>
</tr>
<tr>
<td>5</td>
<td>53</td>
<td>20</td>
</tr>
<tr>
<td>6</td>
<td>54</td>
<td>21</td>
</tr>
<tr>
<td>7</td>
<td>55</td>
<td>22</td>
</tr>
<tr>
<td>8</td>
<td>56</td>
<td>23</td>
</tr>
<tr>
<td>9</td>
<td>57</td>
<td>24</td>
</tr>
<tr>
<td>:</td>
<td>58</td>
<td>25</td>
</tr>
<tr>
<td>;</td>
<td>59</td>
<td>26</td>
</tr>
<tr>
<td>&lt;</td>
<td>60</td>
<td>27</td>
</tr>
<tr>
<td>=</td>
<td>61</td>
<td>28</td>
</tr>
<tr>
<td>&gt;</td>
<td>62</td>
<td>29</td>
</tr>
<tr>
<td>?</td>
<td>63</td>
<td>30</td>
</tr>
<tr>
<td>@</td>
<td>64</td>
<td>31</td>
</tr>
<tr>
<td>A</td>
<td>65</td>
<td>32</td>
</tr>
<tr>
<td>B</td>
<td>66</td>
<td>33</td>
</tr>
<tr>
<td>C</td>
<td>67</td>
<td>34</td>
</tr>
<tr>
<td>D</td>
<td>68</td>
<td>35</td>
</tr>
<tr>
<td>E</td>
<td>69</td>
<td>36</td>
</tr>
<tr>
<td>F</td>
<td>70</td>
<td>37</td>
</tr>
<tr>
<td>G</td>
<td>71</td>
<td>38</td>
</tr>
<tr>
<td>H</td>
<td>72</td>
<td>39</td>
</tr>
<tr>
<td>I</td>
<td>73</td>
<td>40</td>
</tr>
</tbody>
</table>
</details>
<h3 id="assessing-quality-with-fastqc">Assessing quality with FastQC<a class="headerlink" href="#assessing-quality-with-fastqc" title="Permanent link">&para;</a></h3>
<p>The <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> tool provides a
simple way to examine quality metrics for raw sequence data coming from high throughput
sequencing platforms. It provides a modular set of analyses which you can use to give a
quick impression of whether your data has any problems of which you should be aware before
doing any further analysis.</p>
<p>The main functions of FastQC are:</p>
<ul class="task-list">
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> Import of data from BAM, SAM or FastQ files (any variant)</li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> Providing a quick overview to tell you in which areas there may be problems</li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> Summary graphs and tables to quickly assess your data</li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> Export of results to an HTML based permanent report</li>
</ul>
<h3 id="quick-discussion-of-quality-metrics-assessed-by-fastqc">Quick discussion of quality metrics assessed by FastQC<a class="headerlink" href="#quick-discussion-of-quality-metrics-assessed-by-fastqc" title="Permanent link">&para;</a></h3>
<p>For several online discussions about Fastq quality metrics, see</p>
<ul>
<li><a href="https://hbctraining.github.io/Intro-to-rnaseq-hpc-salmon-flipped/lessons/07_qc_fastqc_assessment.html" target="_blank">hbctraining.github.io</a></li>
<li><a href="http://bioinfo-core.org/index.php/9th_Discussion-28_October_2010" target="_blank">ancient but still interesing discussion from the BIOINFO-CORE teal (2010)</a></li>
</ul>
<details class="info">
<summary>Basic Statistics... are basic statitics</summary>
<table>
<thead>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
</thead>
<tbody>
<tr>
<td>Filename</td>
<td>GCB_Mg_S12_forward.gz</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>7998924</td>
</tr>
<tr>
<td>Total Bases</td>
<td>1.2 Gbp</td>
</tr>
<tr>
<td>Sequences flagged as poor quality</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>151</td>
</tr>
<tr>
<td>%GC</td>
<td>48</td>
</tr>
</tbody>
</table>
</details>
<details class="info">
<summary>Per base sequence quality</summary>
<p>is expected to be at minimum of ~30 (.001 probability of wrong base calling)</p>
<p><img alt="" src="../images/fastQCscreenshots/per_base_sequence_quality.png" width="800" /></p>
</details>
<details class="info">
<summary>Per tile sequence quality</summary>
<p>is the representation of the mean Quality in function of the position on the
Affymetrix chip. Non homogeneous quality suggest issues during the sequencing
procedure, of low quality chip.</p>
<p><img alt="" src="../images/fastQCscreenshots/Per_tile_sequence_quality.png" width="800" /></p>
</details>
<details class="info">
<summary>Per sequence quality scores</summary>
<p>Shows the distribution of the quality of all sequences. Generally a high pic of
sequences with high quality is observed, whereas a long tail of small fractions of
total number of sequences have low quality.</p>
<p><img alt="" src="../images/fastQCscreenshots/Per_sequence_quality_scores.png" width="800" /></p>
</details>
<details class="info">
<summary><img alt="⚠" class="twemoji" src="https://cdn.jsdelivr.net/gh/jdecked/twemoji@15.0.3/assets/svg/26a0.svg" title=":warning:" /> Per base sequence content</summary>
<p>Shows the % of each of the four nucleotides in function of their position in the read.
Divergent fraction, often at the beginning of the reads may be due to one or several
combined factors:</p>
<ul>
<li>Illumina devices used to use the 5 first sequencing cycles to calibrate their
  fluorescence detector. Adjustment of the sensitivity of the four wave length chanels
  during the first cycles may cause divergent Nucleotides contents. This factor is
  likely minor with the recent Illumina devices.</li>
<li>If a small number of nucleotides belongs to an adapter or sequence primer, A strong
  bias in nucleotide contents is expected.</li>
<li>Nowadays, preparation of library using a transposase to generate short fragments
  (tagmentation) has become a popular procedure. This transposase does not cut completly
  randomly the DNA. Even a slight bias in tagmentation sites may be responsible for
  divergent fractions of nucleotides at the beginning of the reads.</li>
</ul>
<p><img alt="" src="../images/fastQCscreenshots/Per_base_sequence_content.png" width="800" /></p>
</details>
<details class="info">
<summary>Per sequence GC content</summary>
<p>The distribution of the GC content of sequencing read is in general remarkably stable
for a given organism.
Divergence observed from the theoritical distribution may indicate an overrepresentation
of a few sequences such as ribosomal RNAs or tRNAs (as is the case in the exemple below).</p>
<p>Contamination of samples by another
organism coming either from the experiment or from the experiment of another researcher
may also be the cause of divergence from the theoritical distribution.</p>
<p><img alt="" src="../images/fastQCscreenshots/Per_sequence_GC_content.png" width="800" /></p>
</details>
<details class="info">
<summary>Per base N content</summary>
<p>is expected to be low, otherwise there must have been issues during the base calling
by the illumina device.</p>
</details>
<details class="info">
<summary>Sequence Length Distribution</summary>
<p>is generally an unimodal curve with a pic at the lenght of the reads provided by the
platform.</p>
</details>
<details class="info">
<summary><img alt="⚠" class="twemoji" src="https://cdn.jsdelivr.net/gh/jdecked/twemoji@15.0.3/assets/svg/26a0.svg" title=":warning:" /> Sequence Duplication Levels</summary>
<p>is an important metrics since some analysts are advising to remove PCR duplicates from
sequencing datasets before perfoming read counting. Althought we must keep in mind that
identical molecules (same sequence and same length) may still be biological duplicates
(Think about tRNAs or miRNAs for instance), the advice seems holding for mRNAs.</p>
<p>If you plan to remove "PCR duplicates", this plot allows you to anticipate how usable
reads will be available for RNA profiling after de-duplications.</p>
<p>High duplication levels may be due to the presence of highly expressed RNAs as mentionned
above, or to over PCR-amplification of the library.</p>
<p><img alt="" src="../images/fastQCscreenshots/Sequence_Duplication_Levels.png" width="800" /></p>
</details>
<details class="info">
<summary>Overrepresented sequences</summary>
<p>Are indicative of the presence of contaminating adapters, rRNA, tRNA, etc.</p>
</details>
<details class="info">
<summary>Adapter Content</summary>
<p>Provide the detection of know adapter sequences, generally (but not always) next to
the ends of reads.</p>
<p><img alt="" src="../images/fastQCscreenshots/Adapter_Content.png" width="800" /></p>
</details>
<h2 id="read-filtering">Read filtering<a class="headerlink" href="#read-filtering" title="Permanent link">&para;</a></h2>
<p>It is tempting to <strong>filter</strong> the data to get “good reads” and <strong>discard</strong> "bad reads"
including: </p>
<ul>
<li>Reads with low quality alignments</li>
<li>Reads suspected to be PCR duplicates</li>
</ul>
<p><img alt="⚠" class="twemoji" src="https://cdn.jsdelivr.net/gh/jdecked/twemoji@15.0.3/assets/svg/26a0.svg" title=":warning:" /> <strong>HOWEVER</strong></p>
<p><mark>Discarding reads is a radical decision because it changes the counts</mark></p>
<ul class="task-list">
<li class="task-list-item">
<p><label class="task-list-control"><input type="checkbox" disabled/><span class="task-list-indicator"></span></label> Why low quality reads should be skipped if they were aligned ?</p>
<p>In the case of RNA profiling, we definitely do not advise to remove reads with low
  quality, <em>unless</em> this low quality impacts significantly the number of read effectively
  aligned to the reference genome. Generally, low quality reads are not randomly distributed,
  but, for instance, associated with genes with high GC content. In this case, removing
  low quality reads may skew your counts in favor of AT rich transcripts.</p>
</li>
<li class="task-list-item">
<p><label class="task-list-control"><input type="checkbox" disabled/><span class="task-list-indicator"></span></label> When we remove PCR duplicates (exact same sequence and exact same location), we are
      never 100% sure that we are removing real PCR duplicates, unless a specific bar coding
      system was used (which is done in single-cell RNAseq protocols).</p>
<p>Of course, over amplification of libraries must be corrected by removing duplicated
  sequences. But this can only be done after careful examination of the metrics to
  support this diagnosis.</p>
<p>On another line, ultra deep sequencing of a small genome (or its RNAs) will inevitably cause
  the apparition of duplicated sequences. We can view this trend as a "saturation process",
  well known of geneticist. It may be difficult to deconvolve this expected trend
  from bias due to over amplification of libraries.</p>
</li>
</ul>












                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var tab,labels=set.querySelector(".tabbed-labels");for(tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
    
      
      <nav class="md-footer__inner md-grid" aria-label="Footer" >
        
          
          <a href="../05_sequencing_strategies/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Inserts and sequencing strategies">
            <div class="md-footer__button md-icon">
              
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
            </div>
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Previous
              </span>
              <div class="md-ellipsis">
                Inserts and sequencing strategies
              </div>
            </div>
          </a>
        
        
          
          <a href="../07_transcript_quant/" class="md-footer__link md-footer__link--next" aria-label="Next: Transcript Quantification">
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Next
              </span>
              <div class="md-ellipsis">
                Transcript Quantification
              </div>
            </div>
            <div class="md-footer__button md-icon">
              
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg>
            </div>
          </a>
        
      </nav>
    
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "../..", "features": ["header.autohide", "navigation.instant", "navigation.tracking", "navigation.tabs", "navigation.tabs.sticky", "navigation.top", "navigation.prune", "navigation.footer", "content.tabs.link", "content.code.copy", "content.action.view", "content.action.edit"], "search": "../../assets/javascripts/workers/search.b8dbb3d2.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.1e8ae164.min.js"></script>
      
        <script src="../../javascripts/mathjax.js"></script>
      
        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
      
        <script src="../../spoiler_files/spoiler.js"></script>
      
    
  </body>
</html>