## Become more familiar with scRNAseq analysis thanks to these tutorials

Here are some recommandations on learning how to perform scRNAseq analysis. You can find some tutorials on Galaxy or on R.

### Galaxy Environment
If you want to use the Galaxy environment but you don't know how to use it, we recommend [the following tutorials](https://training.galaxyproject.org/training-material/topics/introduction/)

- *Introduction to Galaxy*
- *A short introduction to Galaxy*
- *Options for using Galaxy*

Then to go further on the scRNAseq analysis, you can perform [these tutorials](https://training.galaxyproject.org/training-material/topics/transcriptomics/)  

- *Introduction to Transcriptomics* (to get in touch with the topic)
- *Understanding Barcodes* & *Plates, Batches, and Barcodes* (to better understand the single cell specificities)
- *Pre-processing of Single-Cell RNA Data* or if you use the 10x genomics technologies : *Pre-processing of 10X Single-Cell RNA Datasets*
- *Single-cell quality control with scater*
- *Downstream Single-cell RNA analysis with RaceID*
- *Clustering 3K PBMCs with Scanpy*

### Rstudio
If you prefer to perform your analysis with R, there is a great tutorial of the package Seurat :  

- [Whole workflow](https://satijalab.org/seurat/v3.0/pbmc3k_tutorial.html)
- [Better understanding of cell-cycle regression](https://satijalab.org/seurat/v3.0/cell_cycle_vignette.html)
