Here are some suggestions to learn how performing scRNAseq analysis, using Galaxy
or R environments.

## Galaxy

If you are new with the Galaxy environment, we recommend
[the following tutorials](https://training.galaxyproject.org/training-material/topics/introduction/)

- [Introduction to Galaxy](https://training.galaxyproject.org/training-material/topics/introduction/slides/introduction.html#1)
  (Use ++up++ and ++down++ keys to scroll slides)
- [A short introduction to Galaxy](https://training.galaxyproject.org/training-material/topics/introduction/tutorials/galaxy-intro-short/tutorial.html)
- [Options for using Galaxy](https://training.galaxyproject.org/training-material/topics/introduction/tutorials/options-for-using-galaxy/slides.html)
  (Use ++up++ and ++down++ keys to scroll slides)

Then to go further on scRNAseq analysis, you can follow
[these tutorials](https://training.galaxyproject.org/training-material/topics/transcriptomics/)  

- [Introduction to Transcriptomics](https://training.galaxyproject.org/training-material/topics/transcriptomics/slides/introduction.html#1)
  (Use ++up++ and ++down++ keys to scroll slides)
- [Plates, Batches, and Barcodes](https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/scrna-plates-batches-barcodes/slides.html)
  (Use ++up++ and ++down++ keys to scroll slides)
- [Understanding Barcodes](https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/scrna-umis/tutorial.html)
- Pre-processing of Single-Cell RNA Data
    - [Slides](https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/scrna-preprocessing/slides.html)
      (Use ++up++ and ++down++ keys to scroll slides)
    - [Tutorials](https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/scrna-preprocessing/tutorial.html)
- [Pre-processing of 10X Single-Cell RNA Datasets](https://training.galaxyproject.org/training-material/topics/transcriptomics/#)
- [Single-cell quality control with scater](https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/scrna-scater-qc/tutorial.html)
- [Downstream Single-cell RNA analysis with RaceID](https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/scrna-raceid/tutorial.html)
- Clustering 3K PBMCs with Scanpy
    - [Slides](https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/scrna-scanpy-pbmc3k/slides.html)
    - [Tutorial](https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/scrna-scanpy-pbmc3k/tutorial.html)

## Analysis in R (or Rstudio)

A great tutorial to the Seurat package  :  

- [Whole workflow](https://satijalab.org/seurat/v3.0/pbmc3k_tutorial.html)
- [Better understanding of cell-cycle regression](https://satijalab.org/seurat/v3.0/cell_cycle_vignette.html)
