![](images/galaxylogo.png)

# `featureCounts`

----
![](images/tool_small.png)

  1. In your history `HISAT2` or `STAR`
  3. Select the `featureCounts` tool with the following parameters to count your reads:
      1. `Alignment file`: select multiple datasets button and shift-click the 7 bam files you have generated
      2. `Specify strand information`: **Unstranded**
      3. `Gene annotation file` : **in your history**
          - `Gene annotation file`: **Drosophila_melanogaster.BDGP6.95.gtf**
      4. `FASTA/Q file`: **Gene-ID "\t" read-count (MultiQC/DESeq2/edgeR/limma-voom compatible)**
      5. `Create gene-length file`: Yes
      6. In `Options for paired-end reads`:
          - `Count fragments instead of reads`: **Enabled; fragments (or templates) will be counted instead of reads**
      7. In `Advanced options`:
          - `GFF feature type filter`: **exon**
          - `GFF gene identifier`: gene_id
          - `Allow read to contribute to multiple features`: No
          - `Count multi-mapping reads/fragments`: Disabled; multi-mapping reads are excluded (default)
          - `Minimum mapping quality per read`: 10
      8. Leave other settings as defaults
  3. `Execute`
  
----
![](images/rename_datasets.png)

You need now to rename you datasets to facilitate your downstream analysis.

Be quiet and focus ! No hurry, this is an *important task* in the analysis.

1. Search and select datasets with `featurecounts` (as we did before for renaming datasets)

2. Click on the info icon ![](images/info.png) of `featureCounts on xxx: Counts`

3. Copy the name of the dataset `Alignment file` in the tool parameters table (for instance, `GSM461176_untreat_single.bam`)
4. Now click on the pencil icon of the *_same_* dataset
5. Paste your text in the `Name` field of the dataset
6. Edit your text by replacing `bam` by `Counts`
(e.g. GSM461176_untreat_single.Counts)
8. repeat _ad lib_ for all counts files generated by featureCounts

# `MultiQC`

We have now generated (1) Bam alignments and (2) Counts files with feature counts, and
we have carefully and courageously edited the names of generated datasets. We are going
to be rewarded for this effort in the next steps !

----
![](images/tool_small.png)

  1. In your history `HISAT2` or `STAR`
  2. Select the `MultiQC` tool with the following parameters:
      1. `1: Results`/ `Which tool was used generate logs?`: **STAR** `or` **HISAT2** (depending on your analysis track)
      2. `STAR or HISAT output`: shift-click select all files with the extension **.log**
      3. Click on the `+ Insert Result` button
      4. `2: Results`/ `Which tool was used generate logs?`: **featureCounts**
      5. `Output of FeatureCounts`: shift-click select all files with the extension **: Summary**
  3. `Execute`

----
![](images/oeil.png) Examine the results and ![](images/lamp.png)




